<?php

    include '../../../common/include/common.php';
    include_once '../../config.php';
    $module = modules::GATK;
    printHeader($module, "Picard | Tool Documentation Index", "Guide");
?>

<link type='text/css' rel='stylesheet' href='picardDoc.css'>

<div class='row-fluid' id="top">

	

	<?php $group = 'Base Calling'; ?>

	<section class="span4">
		<aside class="well">
			<a href="index.html"><h4><i class='fa fa-chevron-left'></i> Back to Tool Docs Index</h4></a>
		</aside>
		<aside class="well">
			<h2>Categories</h2>
        <style>
            #sidenav .accordion-body a {
                color : gray;
            }

            .accordion-body li {
                list-style : none;
            }
        </style>
        <ul class="nav nav-pills nav-stacked" id="sidenav">
        		<hr>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#BaseCalling">Base Calling</a>
							<div id="BaseCalling"
								<?php echo ($group == 'Base Calling')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_illumina_CheckIlluminaDirectory.html">CheckIlluminaDirectory</a>
											</li>
											<li>
												<a href="picard_illumina_CollectIlluminaBasecallingMetrics.html">CollectIlluminaBasecallingMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_CollectIlluminaLaneMetrics.html">CollectIlluminaLaneMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_ExtractIlluminaBarcodes.html">ExtractIlluminaBarcodes</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaBasecallsToFastq.html">IlluminaBasecallsToFastq</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaBasecallsToSam.html">IlluminaBasecallsToSam</a>
											</li>
											<li>
												<a href="picard_illumina_MarkIlluminaAdapters.html">MarkIlluminaAdapters</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#DiagnosticsandQualityControl">Diagnostics and Quality Control</a>
							<div id="DiagnosticsandQualityControl"
								<?php echo ($group == 'Diagnostics and Quality Control')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_AccumulateQualityYieldMetrics.html">AccumulateQualityYieldMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_AccumulateVariantCallingMetrics.html">AccumulateVariantCallingMetrics</a>
											</li>
											<li>
												<a href="picard_sam_BamIndexStats.html">BamIndexStats</a>
											</li>
											<li>
												<a href="picard_fingerprint_CalculateFingerprintMetrics.html">CalculateFingerprintMetrics</a>
											</li>
											<li>
												<a href="picard_sam_CalculateReadGroupChecksum.html">CalculateReadGroupChecksum</a>
											</li>
											<li>
												<a href="picard_fingerprint_CheckFingerprint.html">CheckFingerprint</a>
											</li>
											<li>
												<a href="picard_sam_CheckTerminatorBlock.html">CheckTerminatorBlock</a>
											</li>
											<li>
												<a href="picard_fingerprint_ClusterCrosscheckMetrics.html">ClusterCrosscheckMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectAlignmentSummaryMetrics.html">CollectAlignmentSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_CollectArraysVariantCallingMetrics.html">CollectArraysVariantCallingMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectBaseDistributionByCycle.html">CollectBaseDistributionByCycle</a>
											</li>
											<li>
												<a href="picard_analysis_CollectGcBiasMetrics.html">CollectGcBiasMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_quality_CollectHiSeqXPfFailMetrics.html">CollectHiSeqXPfFailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_directed_CollectHsMetrics.html">CollectHsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_replicates_CollectIndependentReplicateMetrics.html">CollectIndependentReplicateMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectInsertSizeMetrics.html">CollectInsertSizeMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectJumpingLibraryMetrics.html">CollectJumpingLibraryMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectMultipleMetrics.html">CollectMultipleMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectOxoGMetrics.html">CollectOxoGMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectQualityYieldMetrics.html">CollectQualityYieldMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRawWgsMetrics.html">CollectRawWgsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRnaSeqMetrics.html">CollectRnaSeqMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRrbsMetrics.html">CollectRrbsMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_CollectSamErrorMetrics.html">CollectSamErrorMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_CollectSequencingArtifactMetrics.html">CollectSequencingArtifactMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_directed_CollectTargetedPcrMetrics.html">CollectTargetedPcrMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_CollectVariantCallingMetrics.html">CollectVariantCallingMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectWgsMetrics.html">CollectWgsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectWgsMetricsWithNonZeroCoverage.html">CollectWgsMetricsWithNonZeroCoverage</a>
											</li>
											<li>
												<a href="picard_analysis_CompareMetrics.html">CompareMetrics</a>
											</li>
											<li>
												<a href="picard_sam_CompareSAMs.html">CompareSAMs</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_ConvertSequencingArtifactToOxoG.html">ConvertSequencingArtifactToOxoG</a>
											</li>
											<li>
												<a href="picard_fingerprint_CrosscheckFingerprints.html">CrosscheckFingerprints</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_EstimateLibraryComplexity.html">EstimateLibraryComplexity</a>
											</li>
											<li>
												<a href="picard_analysis_MeanQualityByCycle.html">MeanQualityByCycle</a>
											</li>
											<li>
												<a href="picard_analysis_QualityScoreDistribution.html">QualityScoreDistribution</a>
											</li>
											<li>
												<a href="picard_sam_ValidateSamFile.html">ValidateSamFile</a>
											</li>
											<li>
												<a href="picard_sam_ViewSam.html">ViewSam</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#GenotypingArraysManipulation">Genotyping Arrays Manipulation</a>
							<div id="GenotypingArraysManipulation"
								<?php echo ($group == 'Genotyping Arrays Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_arrays_illumina_BpmToNormalizationManifestCsv.html">BpmToNormalizationManifestCsv</a>
											</li>
											<li>
												<a href="picard_arrays_CombineGenotypingArrayVcfs.html">CombineGenotypingArrayVcfs</a>
											</li>
											<li>
												<a href="picard_arrays_CreateBafRegressMetricsFile.html">CreateBafRegressMetricsFile</a>
											</li>
											<li>
												<a href="picard_arrays_CreateVerifyIDIntensityContaminationMetricsFile.html">CreateVerifyIDIntensityContaminationMetricsFile</a>
											</li>
											<li>
												<a href="picard_arrays_GtcToVcf.html">GtcToVcf</a>
											</li>
											<li>
												<a href="picard_arrays_MergePedIntoVcf.html">MergePedIntoVcf</a>
											</li>
											<li>
												<a href="picard_arrays_VcfToAdpc.html">VcfToAdpc</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#IntervalsManipulation">Intervals Manipulation</a>
							<div id="IntervalsManipulation"
								<?php echo ($group == 'Intervals Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_BedToIntervalList.html">BedToIntervalList</a>
											</li>
											<li>
												<a href="picard_util_IntervalListToBed.html">IntervalListToBed</a>
											</li>
											<li>
												<a href="picard_util_IntervalListTools.html">IntervalListTools</a>
											</li>
											<li>
												<a href="picard_util_LiftOverIntervalList.html">LiftOverIntervalList</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Other">Other</a>
							<div id="Other"
								<?php echo ($group == 'Other')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_FifoBuffer.html">FifoBuffer</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReadDataManipulation">Read Data Manipulation</a>
							<div id="ReadDataManipulation"
								<?php echo ($group == 'Read Data Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_sam_AddCommentsToBam.html">AddCommentsToBam</a>
											</li>
											<li>
												<a href="picard_sam_AddOATag.html">AddOATag</a>
											</li>
											<li>
												<a href="picard_sam_AddOrReplaceReadGroups.html">AddOrReplaceReadGroups</a>
											</li>
											<li>
												<a href="picard_fastq_BamToBfq.html">BamToBfq</a>
											</li>
											<li>
												<a href="picard_sam_BuildBamIndex.html">BuildBamIndex</a>
											</li>
											<li>
												<a href="picard_sam_CleanSam.html">CleanSam</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_CollectDuplicateMetrics.html">CollectDuplicateMetrics</a>
											</li>
											<li>
												<a href="picard_sam_DownsampleSam.html">DownsampleSam</a>
											</li>
											<li>
												<a href="picard_sam_FastqToSam.html">FastqToSam</a>
											</li>
											<li>
												<a href="picard_sam_FilterSamReads.html">FilterSamReads</a>
											</li>
											<li>
												<a href="picard_sam_FixMateInformation.html">FixMateInformation</a>
											</li>
											<li>
												<a href="picard_sam_GatherBamFiles.html">GatherBamFiles</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_MarkDuplicates.html">MarkDuplicates</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_MarkDuplicatesWithMateCigar.html">MarkDuplicatesWithMateCigar</a>
											</li>
											<li>
												<a href="picard_sam_MergeBamAlignment.html">MergeBamAlignment</a>
											</li>
											<li>
												<a href="picard_sam_MergeSamFiles.html">MergeSamFiles</a>
											</li>
											<li>
												<a href="picard_sam_PositionBasedDownsampleSam.html">PositionBasedDownsampleSam</a>
											</li>
											<li>
												<a href="picard_sam_ReorderSam.html">ReorderSam</a>
											</li>
											<li>
												<a href="picard_sam_ReplaceSamHeader.html">ReplaceSamHeader</a>
											</li>
											<li>
												<a href="picard_sam_RevertOriginalBaseQualitiesAndAddMateCigar.html">RevertOriginalBaseQualitiesAndAddMateCigar</a>
											</li>
											<li>
												<a href="picard_sam_RevertSam.html">RevertSam</a>
											</li>
											<li>
												<a href="picard_sam_SamFormatConverter.html">SamFormatConverter</a>
											</li>
											<li>
												<a href="picard_sam_SamToFastq.html">SamToFastq</a>
											</li>
											<li>
												<a href="picard_sam_SetNmAndUqTags.html">SetNmAndUqTags</a>
											</li>
											<li>
												<a href="picard_sam_SetNmMdAndUqTags.html">SetNmMdAndUqTags</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_SimpleMarkDuplicatesWithMateCigar.html">SimpleMarkDuplicatesWithMateCigar</a>
											</li>
											<li>
												<a href="picard_sam_SortSam.html">SortSam</a>
											</li>
											<li>
												<a href="picard_sam_SplitSamByLibrary.html">SplitSamByLibrary</a>
											</li>
											<li>
												<a href="picard_sam_SplitSamByNumberOfReads.html">SplitSamByNumberOfReads</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_UmiAwareMarkDuplicatesWithMateCigar.html">UmiAwareMarkDuplicatesWithMateCigar</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Reference">Reference</a>
							<div id="Reference"
								<?php echo ($group == 'Reference')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_BaitDesigner.html">BaitDesigner</a>
											</li>
											<li>
												<a href="picard_sam_CreateSequenceDictionary.html">CreateSequenceDictionary</a>
											</li>
											<li>
												<a href="picard_reference_ExtractSequences.html">ExtractSequences</a>
											</li>
											<li>
												<a href="picard_reference_NonNFastaSize.html">NonNFastaSize</a>
											</li>
											<li>
												<a href="picard_reference_NormalizeFasta.html">NormalizeFasta</a>
											</li>
											<li>
												<a href="picard_util_ScatterIntervalsByNs.html">ScatterIntervalsByNs</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantEvaluationandRefinement">Variant Evaluation and Refinement</a>
							<div id="VariantEvaluationandRefinement"
								<?php echo ($group == 'Variant Evaluation and Refinement')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_vcf_MendelianViolations_FindMendelianViolations.html">FindMendelianViolations</a>
											</li>
											<li>
												<a href="picard_vcf_GenotypeConcordance.html">GenotypeConcordance</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantFiltering">Variant Filtering</a>
							<div id="VariantFiltering"
								<?php echo ($group == 'Variant Filtering')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_vcf_filter_FilterVcf.html">FilterVcf</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantManipulation">Variant Manipulation</a>
							<div id="VariantManipulation"
								<?php echo ($group == 'Variant Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_vcf_FixVcfHeader.html">FixVcfHeader</a>
											</li>
											<li>
												<a href="picard_vcf_GatherVcfs.html">GatherVcfs</a>
											</li>
											<li>
												<a href="picard_vcf_LiftoverVcf.html">LiftoverVcf</a>
											</li>
											<li>
												<a href="picard_vcf_MakeSitesOnlyVcf.html">MakeSitesOnlyVcf</a>
											</li>
											<li>
												<a href="picard_vcf_MergeVcfs.html">MergeVcfs</a>
											</li>
											<li>
												<a href="picard_vcf_RenameSampleInVcf.html">RenameSampleInVcf</a>
											</li>
											<li>
												<a href="picard_vcf_SortVcf.html">SortVcf</a>
											</li>
											<li>
												<a href="picard_vcf_SplitVcfs.html">SplitVcfs</a>
											</li>
											<li>
												<a href="picard_vcf_UpdateVcfSequenceDictionary.html">UpdateVcfSequenceDictionary</a>
											</li>
											<li>
												<a href="picard_vcf_VcfFormatConverter.html">VcfFormatConverter</a>
											</li>
											<li>
												<a href="picard_vcf_VcfToIntervalList.html">VcfToIntervalList</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Metrics">Metrics</a>
							<div id="Metrics"
								<?php echo ($group == 'Metrics')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_analysis_AlignmentSummaryMetrics.html">AlignmentSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_CollectArraysVariantCallingMetrics$ArraysControlCodesSummaryMetrics.html">ArraysControlCodesSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_CollectArraysVariantCallingMetrics$ArraysVariantCallingDetailMetrics.html">ArraysVariantCallingDetailMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_CollectArraysVariantCallingMetrics$ArraysVariantCallingSummaryMetrics.html">ArraysVariantCallingSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_BafRegressMetrics.html">BafRegressMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_SequencingArtifactMetrics$BaitBiasDetailMetrics.html">BaitBiasDetailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_SequencingArtifactMetrics$BaitBiasSummaryMetrics.html">BaitBiasSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_BarcodeMetric.html">BarcodeMetric</a>
											</li>
											<li>
												<a href="picard_analysis_BaseDistributionByCycleMetrics.html">BaseDistributionByCycleMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_BaseErrorMetric.html">BaseErrorMetric</a>
											</li>
											<li>
												<a href="picard_fingerprint_ClusteredCrosscheckMetric.html">ClusteredCrosscheckMetric</a>
											</li>
											<li>
												<a href="picard_analysis_CollectOxoGMetrics$CpcgMetrics.html">CpcgMetrics</a>
											</li>
											<li>
												<a href="picard_sam_DuplicationMetrics.html">DuplicationMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_ErrorMetric.html">ErrorMetric</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_ErrorSummaryMetrics.html">ErrorSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_fingerprint_FingerprintMetrics.html">FingerprintMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_FingerprintingDetailMetrics.html">FingerprintingDetailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_FingerprintingSummaryMetrics.html">FingerprintingSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_GcBiasDetailMetrics.html">GcBiasDetailMetrics</a>
											</li>
											<li>
												<a href="picard_metrics_GcBiasMetrics.html">GcBiasMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_GcBiasSummaryMetrics.html">GcBiasSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_GenotypeConcordanceContingencyMetrics.html">GenotypeConcordanceContingencyMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_GenotypeConcordanceDetailMetrics.html">GenotypeConcordanceDetailMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_GenotypeConcordanceSummaryMetrics.html">GenotypeConcordanceSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_directed_HsMetrics.html">HsMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaBasecallingMetrics.html">IlluminaBasecallingMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaLaneMetrics.html">IlluminaLaneMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaPhasingMetrics.html">IlluminaPhasingMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_IndelErrorMetric.html">IndelErrorMetric</a>
											</li>
											<li>
												<a href="picard_analysis_replicates_IndependentReplicateMetric.html">IndependentReplicateMetric</a>
											</li>
											<li>
												<a href="picard_analysis_InsertSizeMetrics.html">InsertSizeMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_JumpingLibraryMetrics.html">JumpingLibraryMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_MendelianViolations_MendelianViolationMetrics.html">MendelianViolationMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_OverlappingErrorMetric.html">OverlappingErrorMetric</a>
											</li>
											<li>
												<a href="picard_illumina_quality_CollectHiSeqXPfFailMetrics$PFFailDetailedMetric.html">PFFailDetailedMetric</a>
											</li>
											<li>
												<a href="picard_illumina_quality_CollectHiSeqXPfFailMetrics$PFFailSummaryMetric.html">PFFailSummaryMetric</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_SequencingArtifactMetrics$PreAdapterDetailMetrics.html">PreAdapterDetailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_SequencingArtifactMetrics$PreAdapterSummaryMetrics.html">PreAdapterSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectQualityYieldMetrics$QualityYieldMetrics.html">QualityYieldMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRawWgsMetrics$RawWgsMetrics.html">RawWgsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_RnaSeqMetrics.html">RnaSeqMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_RrbsCpgDetailMetrics.html">RrbsCpgDetailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_RrbsSummaryMetrics.html">RrbsSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamComparisonMetric.html">SamComparisonMetric</a>
											</li>
											<li>
												<a href="picard_analysis_directed_TargetMetrics.html">TargetMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_directed_TargetedPcrMetrics.html">TargetedPcrMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_TheoreticalSensitivityMetrics.html">TheoreticalSensitivityMetrics</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_UmiMetrics.html">UmiMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_CollectVariantCallingMetrics$VariantCallingDetailMetrics.html">VariantCallingDetailMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_CollectVariantCallingMetrics$VariantCallingSummaryMetrics.html">VariantCallingSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_VerifyIDIntensityContaminationMetrics.html">VerifyIDIntensityContaminationMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_WgsMetrics.html">WgsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectWgsMetricsWithNonZeroCoverage$WgsMetricsWithNonZeroCoverage.html">WgsMetricsWithNonZeroCoverage</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
        </ul>
		</aside>
		<?php getForumPosts( 'IlluminaBasecallsToSam' ) ?>

	</section>

	<div class="span8">

		<h1>IlluminaBasecallsToSam</h1>

		<p class="lead">Transforms raw Illumina sequencing data into an unmapped SAM, BAM or CRAM file.<p>The IlluminaBaseCallsToSam program collects, demultiplexes, and sorts reads across all of the tiles of a lane via barcode to produce an unmapped SAM, BAM or CRAM file.  An unmapped BAM file is often referred to as a uBAM.  All barcode, sample, and library data is provided in the LIBRARY_PARAMS file.  Note, this LIBRARY_PARAMS file should be formatted according to the specifications indicated below.  The following is an example of a properly formatted LIBRARY_PARAMS file:</p>BARCODE_1	OUTPUT	SAMPLE_ALIAS	LIBRARY_NAME
AAAAAAAA	SA_AAAAAAAA.bam	SA_AAAAAAAA	LN_AAAAAAAA
AAAAGAAG	SA_AAAAGAAG.bam	SA_AAAAGAAG	LN_AAAAGAAG
AACAATGG	SA_AACAATGG.bam	SA_AACAATGG	LN_AACAATGG
N	SA_non_indexed.bam	SA_non_indexed	LN_NNNNNNNN
 <p>The BARCODES_DIR file is produced by the <a href='http://broadinstitute.github.io/picard/command-line-overview.html#ExtractIlluminaBarcodes'>ExtractIlluminaBarcodes</a> tool for each lane of a flow cell.</p>  <p>Barcode matching can be done inline without requiring barcodes files generated by `ExtractIlluminaBarcode`. By setting MATCH_BARCODES_INLINE to true barcodes will be matched as they are parsed and converted. Thisdoes not require BARCODES_DIR.</p>     <h4>Usage example:</h4><pre>java -jar picard.jar IlluminaBasecallsToSam \<br />      BASECALLS_DIR=/BaseCalls/ \<br />      LANE=001 \<br />      READ_STRUCTURE=25T8B25T \<br />      RUN_BARCODE=run15 \<br />      IGNORE_UNEXPECTED_BARCODES=true \<br />      LIBRARY_PARAMS=library.params </pre><hr /></p>

			<h3>Category
				<small> Base Calling</small>
			</h3>
		<hr>
		<h2>Overview</h2>
		IlluminaBasecallsToSam transforms a lane of Illumina data file formats (bcl, locs, clocs, qseqs, etc.) into
 SAM, BAM or CRAM file format.
 <p/>
 In this application, barcode data is read from Illumina data file groups, each of which is associated with a tile.
 Each tile may contain data for any number of barcodes, and a single barcode's data may span multiple tiles.  Once the
 barcode data is collected from files, each barcode's data is written to its own SAM/BAM/CRAM.  The barcode data must be
 written in order; this means that barcode data from each tile is sorted before it is written to file, and that if a
 barcode's data does span multiple tiles, data collected from each tile must be written in the order of the tiles
 themselves.
 <p/>
 This class employs a number of private subclasses to achieve this goal.  The TileReadAggregator controls the flow
 of operation.  It is fed a number of Tiles which it uses to spawn TileReaders.  TileReaders are responsible for
 reading Illumina data for their respective tiles from disk, and as they collect that data, it is fed back into the
 TileReadAggregator.  When a TileReader completes a tile, it notifies the TileReadAggregator, which reviews what was
 read and conditionally queues its writing to disk, baring in mind the requirements of write-order described in the
 previous paragraph.  As writes complete, the TileReadAggregator re-evaluates the state of reads/writes and may queue
 more writes.  When all barcodes for all tiles have been written, the TileReadAggregator shuts down.
 <p/>
 The TileReadAggregator controls task execution using a specialized ThreadPoolExecutor.  It accepts special Runnables
 of type PriorityRunnable which allow a priority to be assigned to the runnable.  When the ThreadPoolExecutor is
 assigning threads, it gives priority to those PriorityRunnables with higher priority values.  In this application,
 TileReaders are assigned lowest priority, and write tasks are assigned high priority.  It is designed in this fashion
 to minimize the amount of time data must remain in memory (write the data as soon as possible, then discard it from
 memory) while maximizing CPU usage.
@author jburke@broadinstitute.org
@author mccowan@broadinstitute.org


				<h3>IlluminaBasecallsToSam specific arguments</h3>
				<p>This table summarizes the command-line arguments that are specific to this tool. For more details on each argument, see the list further down below the table or click on an argument name to jump directly to that entry in the list.</p>
				<table class="table table-striped table-bordered table-condensed">
					<thead>
					<tr>
						<th>Argument name(s)</th>
						<th>Default value</th>
						<th>Summary</th>
					</tr>
					</thead>
					<tbody>
			<tr>
				<th colspan="4" id="row-divider">Required Arguments</th>
			</tr>
				<tr>
					<td><a href="#--BARCODE_PARAMS">--BARCODE_PARAMS</a><br />
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>Deprecated (use LIBRARY_PARAMS).  Tab-separated file for creating all output SAM, BAM or CRAM files for barcoded run with single IlluminaBasecallsToSam invocation.  Columns are BARCODE, OUTPUT, SAMPLE_ALIAS, and LIBRARY_NAME.  Row with BARCODE=N is used to specify a file for no barcode match</td>
				</tr>
				<tr>
					<td><a href="#--BASECALLS_DIR">--BASECALLS_DIR</a><br />
								&nbsp;<em>-B</em>
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>The Illumina basecalls directory. </td>
				</tr>
				<tr>
					<td><a href="#--LANE">--LANE</a><br />
								&nbsp;<em>-L</em>
					</td>
					<!--<td>List[Integer]</td> -->
					<td>[]</td>
					<td>Lane number. This can be specified multiple times. Reads with the same index in multiple lanes will be added to the same output file.</td>
				</tr>
				<tr>
					<td><a href="#--LIBRARY_PARAMS">--LIBRARY_PARAMS</a><br />
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>Tab-separated file for creating all output SAM, BAM or CRAM files for a lane with single IlluminaBasecallsToSam invocation.  The columns are OUTPUT, SAMPLE_ALIAS, and LIBRARY_NAME, BARCODE_1, BARCODE_2 ... BARCODE_X where X = number of barcodes per cluster (optional).  Row with BARCODE_1 set to 'N' is used to specify a file for no barcode match.  You may also provide any 2 letter RG header attributes (excluding PU, CN, PL, and DT)  as columns in this file and the values for those columns will be inserted into the RG tag for the SAM, BAM or CRAM file created for a given row.</td>
				</tr>
				<tr>
					<td><a href="#--OUTPUT">--OUTPUT</a><br />
								&nbsp;<em>-O</em>
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>Deprecated (use LIBRARY_PARAMS).  The output SAM, BAM or CRAM file. Format is determined by extension.</td>
				</tr>
				<tr>
					<td><a href="#--READ_STRUCTURE">--READ_STRUCTURE</a><br />
								&nbsp;<em>-RS</em>
					</td>
					<!--<td>String</td> -->
					<td>null</td>
					<td>A description of the logical structure of clusters in an Illumina Run, i.e. a description of the structure IlluminaBasecallsToSam assumes the  data to be in. It should consist of integer/character pairs describing the number of cycles and the type of those cycles (B for Sample Barcode, M for molecular barcode, T for Template, and S for skip).  E.g. If the input data consists of 80 base clusters and we provide a read structure of "28T8M8B8S28T" then the sequence may be split up into four reads:
* read one with 28 cycles (bases) of template
* read two with 8 cycles (bases) of molecular barcode (ex. unique molecular barcode)
* read three with 8 cycles (bases) of sample barcode
* 8 cycles (bases) skipped.
* read four with 28 cycles (bases) of template
The skipped cycles would NOT be included in an output SAM/BAM file or in read groups therein.</td>
				</tr>
				<tr>
					<td><a href="#--RUN_BARCODE">--RUN_BARCODE</a><br />
					</td>
					<!--<td>String</td> -->
					<td>null</td>
					<td>The barcode of the run.  Prefixed to read names.</td>
				</tr>
				<tr>
					<td><a href="#--SAMPLE_ALIAS">--SAMPLE_ALIAS</a><br />
								&nbsp;<em>-ALIAS</em>
					</td>
					<!--<td>String</td> -->
					<td>null</td>
					<td>Deprecated (use LIBRARY_PARAMS).  The name of the sequenced sample</td>
				</tr>
				<tr>
					<td><a href="#--SEQUENCING_CENTER">--SEQUENCING_CENTER</a><br />
					</td>
					<!--<td>String</td> -->
					<td>null</td>
					<td>The name of the sequencing center that produced the reads.  Used to set the @RG->CN header tag.</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Tool Arguments</th>
			</tr>
				<tr>
					<td><a href="#--ADAPTERS_TO_CHECK">--ADAPTERS_TO_CHECK</a><br />
					</td>
					<!--<td>List[IlluminaAdapterPair]</td> -->
					<td>[INDEXED, DUAL_INDEXED, NEXTERA_V2, FLUIDIGM]</td>
					<td>Which adapters to look for in the read.</td>
				</tr>
				<tr>
					<td><a href="#--APPLY_EAMSS_FILTER">--APPLY_EAMSS_FILTER</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>true</td>
					<td>Apply EAMSS filtering to identify inappropriately quality scored bases towards the ends of reads and convert their quality scores to Q2.</td>
				</tr>
				<tr>
					<td><a href="#--arguments_file">--arguments_file</a><br />
					</td>
					<!--<td>List[File]</td> -->
					<td>[]</td>
					<td>read one or more arguments files and add them to the command line</td>
				</tr>
				<tr>
					<td><a href="#--BARCODE_POPULATION_STRATEGY">--BARCODE_POPULATION_STRATEGY</a><br />
					</td>
					<!--<td>PopulateBarcode</td> -->
					<td>ORPHANS_ONLY</td>
					<td>When should the sample barcode (as read by the sequencer) be placed on the reads in the BC tag?</td>
				</tr>
				<tr>
					<td><a href="#--BARCODES_DIR">--BARCODES_DIR</a><br />
								&nbsp;<em>-BCD</em>
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>The barcodes directory with _barcode.txt files (generated by ExtractIlluminaBarcodes). If not set, use BASECALLS_DIR. </td>
				</tr>
				<tr>
					<td><a href="#--COMPRESS_OUTPUTS">--COMPRESS_OUTPUTS</a><br />
								&nbsp;<em>-GZIP</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Compress output FASTQ files using gzip and append a .gz extension to the file names.</td>
				</tr>
				<tr>
					<td><a href="#--DISTANCE_MODE">--DISTANCE_MODE</a><br />
					</td>
					<!--<td>DistanceMetric</td> -->
					<td>HAMMING</td>
					<td>The distance metric that should be used to compare the barcode-reads and the provided barcodes for finding the best and second-best assignments.</td>
				</tr>
				<tr>
					<td><a href="#--FIRST_TILE">--FIRST_TILE</a><br />
					</td>
					<!--<td>Integer</td> -->
					<td>null</td>
					<td>If set, this is the first tile to be processed (used for debugging).  Note that tiles are not processed in numerical order.</td>
				</tr>
				<tr>
					<td><a href="#--FIVE_PRIME_ADAPTER">--FIVE_PRIME_ADAPTER</a><br />
					</td>
					<!--<td>String</td> -->
					<td>null</td>
					<td>For specifying adapters other than standard Illumina</td>
				</tr>
				<tr>
					<td><a href="#--help">--help</a><br />
								&nbsp;<em>-h</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>display the help message</td>
				</tr>
				<tr>
					<td><a href="#--IGNORE_UNEXPECTED_BARCODES">--IGNORE_UNEXPECTED_BARCODES</a><br />
								&nbsp;<em>-IGNORE_UNEXPECTED</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Whether to ignore reads whose barcodes are not found in LIBRARY_PARAMS.  Useful when outputting SAM, BAM or CRAM files for only a subset of the barcodes in a lane.</td>
				</tr>
				<tr>
					<td><a href="#--INCLUDE_BARCODE_QUALITY">--INCLUDE_BARCODE_QUALITY</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Should the barcode quality be included when the sample barcode is included?</td>
				</tr>
				<tr>
					<td><a href="#--INCLUDE_BC_IN_RG_TAG">--INCLUDE_BC_IN_RG_TAG</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Whether to include the barcode information in the @RG->BC header tag. Defaults to false until included in the SAM spec.</td>
				</tr>
				<tr>
					<td><a href="#--INCLUDE_NON_PF_READS">--INCLUDE_NON_PF_READS</a><br />
								&nbsp;<em>-NONPF</em>
					</td>
					<!--<td>boolean</td> -->
					<td>true</td>
					<td>Whether to include non-PF reads</td>
				</tr>
				<tr>
					<td><a href="#--INPUT_PARAMS_FILE">--INPUT_PARAMS_FILE</a><br />
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>The input file that defines parameters for the program. This is the BARCODE_FILE for `ExtractIlluminaBarcodes` or the MULTIPLEX_PARAMS or LIBRARY_PARAMS file for `IlluminaBasecallsToFastq`  or `IlluminaBasecallsToSam`</td>
				</tr>
				<tr>
					<td><a href="#--LIBRARY_NAME">--LIBRARY_NAME</a><br />
								&nbsp;<em>-LIB</em>
					</td>
					<!--<td>String</td> -->
					<td>null</td>
					<td>Deprecated (use LIBRARY_PARAMS).  The name of the sequenced library</td>
				</tr>
				<tr>
					<td><a href="#--MATCH_BARCODES_INLINE">--MATCH_BARCODES_INLINE</a><br />
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>If true, match barcodes on the fly. Otherwise parse the barcodes from the barcodes file.</td>
				</tr>
				<tr>
					<td><a href="#--MAX_MISMATCHES">--MAX_MISMATCHES</a><br />
					</td>
					<!--<td>int</td> -->
					<td>1</td>
					<td>Maximum mismatches for a barcode to be considered a match.</td>
				</tr>
				<tr>
					<td><a href="#--MAX_NO_CALLS">--MAX_NO_CALLS</a><br />
					</td>
					<!--<td>int</td> -->
					<td>2</td>
					<td>Maximum allowable number of no-calls in a barcode read before it is considered unmatchable.</td>
				</tr>
				<tr>
					<td><a href="#--MAX_READS_IN_RAM_PER_TILE">--MAX_READS_IN_RAM_PER_TILE</a><br />
					</td>
					<!--<td>int</td> -->
					<td>-1</td>
					<td>Configure SortingCollections to store this many records before spilling to disk. For an indexed run, each SortingCollection gets this value/number of indices. Deprecated: use `MAX_RECORDS_IN_RAM`</td>
				</tr>
				<tr>
					<td><a href="#--METRICS_FILE">--METRICS_FILE</a><br />
								&nbsp;<em>-M</em>
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>Per-barcode and per-lane metrics written to this file.</td>
				</tr>
				<tr>
					<td><a href="#--MIN_MISMATCH_DELTA">--MIN_MISMATCH_DELTA</a><br />
					</td>
					<!--<td>int</td> -->
					<td>1</td>
					<td>Minimum difference between number of mismatches in the best and second best barcodes for a barcode to be considered a match.</td>
				</tr>
				<tr>
					<td><a href="#--MINIMUM_BASE_QUALITY">--MINIMUM_BASE_QUALITY</a><br />
								&nbsp;<em>-Q</em>
					</td>
					<!--<td>int</td> -->
					<td>0</td>
					<td>Minimum base quality. Any barcode bases falling below this quality will be considered a mismatch even if the bases match.</td>
				</tr>
				<tr>
					<td><a href="#--MINIMUM_QUALITY">--MINIMUM_QUALITY</a><br />
					</td>
					<!--<td>int</td> -->
					<td>2</td>
					<td>The minimum quality (after transforming 0s to 1s) expected from reads.  If qualities are lower than this value, an error is thrown. The default of 2 is what the Illumina's spec describes as the minimum, but in practice the value has been observed lower.</td>
				</tr>
				<tr>
					<td><a href="#--MOLECULAR_INDEX_BASE_QUALITY_TAG">--MOLECULAR_INDEX_BASE_QUALITY_TAG</a><br />
					</td>
					<!--<td>String</td> -->
					<td>QX</td>
					<td>The tag to use to store any molecular index base qualities.  If more than one molecular index is found, their qualities will be concatenated and stored here (.i.e. the number of "M" operators in the READ_STRUCTURE)</td>
				</tr>
				<tr>
					<td><a href="#--MOLECULAR_INDEX_TAG">--MOLECULAR_INDEX_TAG</a><br />
					</td>
					<!--<td>String</td> -->
					<td>RX</td>
					<td>The tag to use to store any molecular indexes.  If more than one molecular index is found, they will be concatenated and stored here.</td>
				</tr>
				<tr>
					<td><a href="#--NUM_PROCESSORS">--NUM_PROCESSORS</a><br />
					</td>
					<!--<td>Integer</td> -->
					<td>0</td>
					<td>The number of threads to run in parallel. If NUM_PROCESSORS = 0, number of cores is automatically set to the number of cores available on the machine. If NUM_PROCESSORS < 0, then the number of cores used will be the number available on the machine less NUM_PROCESSORS.</td>
				</tr>
				<tr>
					<td><a href="#--PLATFORM">--PLATFORM</a><br />
					</td>
					<!--<td>String</td> -->
					<td>ILLUMINA</td>
					<td>The name of the sequencing technology that produced the read.</td>
				</tr>
				<tr>
					<td><a href="#--PROCESS_SINGLE_TILE">--PROCESS_SINGLE_TILE</a><br />
					</td>
					<!--<td>Integer</td> -->
					<td>null</td>
					<td>If set, process only the tile number given and prepend the tile number to the output file name.</td>
				</tr>
				<tr>
					<td><a href="#--READ_GROUP_ID">--READ_GROUP_ID</a><br />
								&nbsp;<em>-RG</em>
					</td>
					<!--<td>String</td> -->
					<td>null</td>
					<td>ID used to link RG header record with RG tag in SAM record.  If these are unique in SAM files that get merged, merge performance is better.  If not specified, READ_GROUP_ID will be set to <first 5 chars of RUN_BARCODE>.<LANE> .</td>
				</tr>
				<tr>
					<td><a href="#--RUN_START_DATE">--RUN_START_DATE</a><br />
					</td>
					<!--<td>Date</td> -->
					<td>null</td>
					<td>The start date of the run.</td>
				</tr>
				<tr>
					<td><a href="#--SORT">--SORT</a><br />
					</td>
					<!--<td>Boolean</td> -->
					<td>true</td>
					<td>If true, the output records are sorted by read name. Otherwise they are unsorted.</td>
				</tr>
				<tr>
					<td><a href="#--TAG_PER_MOLECULAR_INDEX">--TAG_PER_MOLECULAR_INDEX</a><br />
					</td>
					<!--<td>List[String]</td> -->
					<td>[]</td>
					<td>The list of tags to store each molecular index.  The number of tags should match the number of molecular indexes.</td>
				</tr>
				<tr>
					<td><a href="#--THREE_PRIME_ADAPTER">--THREE_PRIME_ADAPTER</a><br />
					</td>
					<!--<td>String</td> -->
					<td>null</td>
					<td>For specifying adapters other than standard Illumina</td>
				</tr>
				<tr>
					<td><a href="#--TILE_LIMIT">--TILE_LIMIT</a><br />
					</td>
					<!--<td>Integer</td> -->
					<td>null</td>
					<td>If set, process no more than this many tiles (used for debugging).</td>
				</tr>
				<tr>
					<td><a href="#--version">--version</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>display the version number for this tool</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Common Arguments</th>
			</tr>
				<tr>
					<td><a href="#--COMPRESSION_LEVEL">--COMPRESSION_LEVEL</a><br />
					</td>
					<!--<td>int</td> -->
					<td>5</td>
					<td>Compression level for all compressed files created (e.g. BAM and VCF).</td>
				</tr>
				<tr>
					<td><a href="#--CREATE_INDEX">--CREATE_INDEX</a><br />
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Whether to create an index when writing VCF or coordinate sorted BAM output.</td>
				</tr>
				<tr>
					<td><a href="#--CREATE_MD5_FILE">--CREATE_MD5_FILE</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Whether to create an MD5 digest for any BAM or FASTQ files created.  </td>
				</tr>
				<tr>
					<td><a href="#--MAX_RECORDS_IN_RAM">--MAX_RECORDS_IN_RAM</a><br />
					</td>
					<!--<td>Integer</td> -->
					<td>500000</td>
					<td>When writing files that need to be sorted, this will specify the number of records stored in RAM before spilling to disk. Increasing this number reduces the number of file handles needed to sort the file, and increases the amount of RAM needed.</td>
				</tr>
				<tr>
					<td><a href="#--QUIET">--QUIET</a><br />
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Whether to suppress job-summary info on System.err.</td>
				</tr>
				<tr>
					<td><a href="#--REFERENCE_SEQUENCE">--REFERENCE_SEQUENCE</a><br />
								&nbsp;<em>-R</em>
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>Reference sequence file.</td>
				</tr>
				<tr>
					<td><a href="#--TMP_DIR">--TMP_DIR</a><br />
					</td>
					<!--<td>List[File]</td> -->
					<td>[]</td>
					<td>One or more directories with space available to be used by this program for temporary storage of working files</td>
				</tr>
				<tr>
					<td><a href="#--USE_JDK_DEFLATER">--USE_JDK_DEFLATER</a><br />
								&nbsp;<em>-use_jdk_deflater</em>
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Use the JDK Deflater instead of the Intel Deflater for writing compressed output</td>
				</tr>
				<tr>
					<td><a href="#--USE_JDK_INFLATER">--USE_JDK_INFLATER</a><br />
								&nbsp;<em>-use_jdk_inflater</em>
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Use the JDK Inflater instead of the Intel Inflater for reading compressed input</td>
				</tr>
				<tr>
					<td><a href="#--VALIDATION_STRINGENCY">--VALIDATION_STRINGENCY</a><br />
					</td>
					<!--<td>ValidationStringency</td> -->
					<td>STRICT</td>
					<td>Validation stringency for all SAM files read by this program.  Setting stringency to SILENT can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded.</td>
				</tr>
				<tr>
					<td><a href="#--VERBOSITY">--VERBOSITY</a><br />
					</td>
					<!--<td>LogLevel</td> -->
					<td>INFO</td>
					<td>Control verbosity of logging.</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Advanced Arguments</th>
			</tr>
				<tr>
					<td><a href="#--showHidden">--showHidden</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>display hidden arguments</td>
				</tr>
					</tbody>
				</table>

					<h3>Argument details</h3>
					<p>Arguments in this list are specific to this tool. Keep in mind that other arguments are available that are shared with other tools (e.g. command-line GATK arguments); see Inherited arguments above.</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--ADAPTERS_TO_CHECK">--ADAPTERS_TO_CHECK </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Which adapters to look for in the read.</b><br />
			
		</p>
			<p>
				The --ADAPTERS_TO_CHECK argument is an enumerated type (List[IlluminaAdapterPair]), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">PAIRED_END</dt>
					<dd class="enum">The following sequences can be found in
        https://support.illumina.com/content/dam/illumina-support/documents/documentation/chemistry_documentation/experiment-design/illumina-adapter-sequences_1000000002694-01.pdf
        and are protected by the following copyright notice:

        Oligonucleotide sequences (c) 2016 Illumina, Inc. All rights reserved. Derivative works created by
        Illumina customers are authorized for use with Illumina instruments and products only. All other
        uses are strictly prohibited.</dd>
					<dt class="enum">INDEXED</dt>
					<dd class="enum"></dd>
					<dt class="enum">SINGLE_END</dt>
					<dd class="enum"></dd>
					<dt class="enum">NEXTERA_V1</dt>
					<dd class="enum"></dd>
					<dt class="enum">NEXTERA_V2</dt>
					<dd class="enum"></dd>
					<dt class="enum">DUAL_INDEXED</dt>
					<dd class="enum"></dd>
					<dt class="enum">FLUIDIGM</dt>
					<dd class="enum"></dd>
					<dt class="enum">TRUSEQ_SMALLRNA</dt>
					<dd class="enum"></dd>
					<dt class="enum">ALTERNATIVE_SINGLE_END</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">List[IlluminaAdapterPair]</span>
			&nbsp;<span class="label">[INDEXED, DUAL_INDEXED, NEXTERA_V2, FLUIDIGM]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--APPLY_EAMSS_FILTER">--APPLY_EAMSS_FILTER </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Apply EAMSS filtering to identify inappropriately quality scored bases towards the ends of reads and convert their quality scores to Q2.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">true</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--arguments_file">--arguments_file </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>read one or more arguments files and add them to the command line</b><br />
			
		</p>
		<p>
		<span class="label label-info ">List[File]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--BARCODE_PARAMS">--BARCODE_PARAMS </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Deprecated (use LIBRARY_PARAMS).  Tab-separated file for creating all output SAM, BAM or CRAM files for barcoded run with single IlluminaBasecallsToSam invocation.  Columns are BARCODE, OUTPUT, SAMPLE_ALIAS, and LIBRARY_NAME.  Row with BARCODE=N is used to specify a file for no barcode match</b><br />
			
		</p>
			<p><b>Exclusion:</b> This argument cannot be used at the same time as <code>OUTPUT, SAMPLE_ALIAS, LIBRARY_NAME, LIBRARY_PARAMS</code>.</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--BARCODE_POPULATION_STRATEGY">--BARCODE_POPULATION_STRATEGY </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>When should the sample barcode (as read by the sequencer) be placed on the reads in the BC tag?</b><br />
			
		</p>
			<p>
				The --BARCODE_POPULATION_STRATEGY argument is an enumerated type (PopulateBarcode), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">ORPHANS_ONLY</dt>
					<dd class="enum">Put barcodes only into the records that were not assigned to any declared barcode.</dd>
					<dt class="enum">INEXACT_MATCH</dt>
					<dd class="enum">Put barcodes into records for which an exact match with a declared barcode was not found.</dd>
					<dt class="enum">ALWAYS</dt>
					<dd class="enum">Put barcodes into all the records.</dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">PopulateBarcode</span>
			&nbsp;<span class="label">ORPHANS_ONLY</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--BARCODES_DIR">--BARCODES_DIR </a>
			 / <small>-BCD</small>
		</h3>
		<p class="args">
			<b>The barcodes directory with _barcode.txt files (generated by ExtractIlluminaBarcodes). If not set, use BASECALLS_DIR. </b><br />
			
		</p>
		<p>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--BASECALLS_DIR">--BASECALLS_DIR </a>
			 / <small>-B</small>
		</h3>
		<p class="args">
			<b>The Illumina basecalls directory. </b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--COMPRESS_OUTPUTS">--COMPRESS_OUTPUTS </a>
			 / <small>-GZIP</small>
		</h3>
		<p class="args">
			<b>Compress output FASTQ files using gzip and append a .gz extension to the file names.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--COMPRESSION_LEVEL">--COMPRESSION_LEVEL </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Compression level for all compressed files created (e.g. BAM and VCF).</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">5</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--CREATE_INDEX">--CREATE_INDEX </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Whether to create an index when writing VCF or coordinate sorted BAM output.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--CREATE_MD5_FILE">--CREATE_MD5_FILE </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Whether to create an MD5 digest for any BAM or FASTQ files created.  </b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--DISTANCE_MODE">--DISTANCE_MODE </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>The distance metric that should be used to compare the barcode-reads and the provided barcodes for finding the best and second-best assignments.</b><br />
			
		</p>
			<p>
				The --DISTANCE_MODE argument is an enumerated type (DistanceMetric), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">HAMMING</dt>
					<dd class="enum">Hamming distance: The n-th base in the read is compared against the n-th base in the barcode. Unequal bases and low quality bases are considered mismatches. No-call read-bases are not considered mismatches. </dd>
					<dt class="enum">LENIENT_HAMMING</dt>
					<dd class="enum">Leniant Hamming distance: The n-th base in the read is compared against the n-th base in the barcode. Unequal bases are considered mismatches. No-call read-bases, or those with low quality are not considered mismatches.</dd>
					<dt class="enum">FREE</dt>
					<dd class="enum">FREE Metric: A Levenshtein-like metric that performs a simple Smith-Waterman with mismatch, gap open, and gap extend costs all equal to 1. Insertions or deletions at the ends of the read or barcode do not count toward the distance. No-call read-bases, or those with low quality are not considered mismatches.</dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">DistanceMetric</span>
			&nbsp;<span class="label">HAMMING</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--FIRST_TILE">--FIRST_TILE </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>If set, this is the first tile to be processed (used for debugging).  Note that tiles are not processed in numerical order.</b><br />
			
		</p>
			<p><b>Exclusion:</b> This argument cannot be used at the same time as <code>PROCESS_SINGLE_TILE</code>.</p>
		<p>
		<span class="label label-info ">Integer</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--FIVE_PRIME_ADAPTER">--FIVE_PRIME_ADAPTER </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>For specifying adapters other than standard Illumina</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--help">--help </a>
			 / <small>-h</small>
		</h3>
		<p class="args">
			<b>display the help message</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--IGNORE_UNEXPECTED_BARCODES">--IGNORE_UNEXPECTED_BARCODES </a>
			 / <small>-IGNORE_UNEXPECTED</small>
		</h3>
		<p class="args">
			<b>Whether to ignore reads whose barcodes are not found in LIBRARY_PARAMS.  Useful when outputting SAM, BAM or CRAM files for only a subset of the barcodes in a lane.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--INCLUDE_BARCODE_QUALITY">--INCLUDE_BARCODE_QUALITY </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Should the barcode quality be included when the sample barcode is included?</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--INCLUDE_BC_IN_RG_TAG">--INCLUDE_BC_IN_RG_TAG </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Whether to include the barcode information in the @RG->BC header tag. Defaults to false until included in the SAM spec.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--INCLUDE_NON_PF_READS">--INCLUDE_NON_PF_READS </a>
			 / <small>-NONPF</small>
		</h3>
		<p class="args">
			<b>Whether to include non-PF reads</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">true</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--INPUT_PARAMS_FILE">--INPUT_PARAMS_FILE </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>The input file that defines parameters for the program. This is the BARCODE_FILE for `ExtractIlluminaBarcodes` or the MULTIPLEX_PARAMS or LIBRARY_PARAMS file for `IlluminaBasecallsToFastq`  or `IlluminaBasecallsToSam`</b><br />
			
		</p>
		<p>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--LANE">--LANE </a>
			 / <small>-L</small>
		</h3>
		<p class="args">
			<b>Lane number. This can be specified multiple times. Reads with the same index in multiple lanes will be added to the same output file.</b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">List[Integer]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--LIBRARY_NAME">--LIBRARY_NAME </a>
			 / <small>-LIB</small>
		</h3>
		<p class="args">
			<b>Deprecated (use LIBRARY_PARAMS).  The name of the sequenced library</b><br />
			
		</p>
			<p><b>Exclusion:</b> This argument cannot be used at the same time as <code>BARCODE_PARAMS, LIBRARY_PARAMS</code>.</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--LIBRARY_PARAMS">--LIBRARY_PARAMS </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Tab-separated file for creating all output SAM, BAM or CRAM files for a lane with single IlluminaBasecallsToSam invocation.  The columns are OUTPUT, SAMPLE_ALIAS, and LIBRARY_NAME, BARCODE_1, BARCODE_2 ... BARCODE_X where X = number of barcodes per cluster (optional).  Row with BARCODE_1 set to 'N' is used to specify a file for no barcode match.  You may also provide any 2 letter RG header attributes (excluding PU, CN, PL, and DT)  as columns in this file and the values for those columns will be inserted into the RG tag for the SAM, BAM or CRAM file created for a given row.</b><br />
			
		</p>
			<p><b>Exclusion:</b> This argument cannot be used at the same time as <code>OUTPUT, SAMPLE_ALIAS, LIBRARY_NAME, BARCODE_PARAMS</code>.</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MATCH_BARCODES_INLINE">--MATCH_BARCODES_INLINE </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>If true, match barcodes on the fly. Otherwise parse the barcodes from the barcodes file.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MAX_MISMATCHES">--MAX_MISMATCHES </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Maximum mismatches for a barcode to be considered a match.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">1</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MAX_NO_CALLS">--MAX_NO_CALLS </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Maximum allowable number of no-calls in a barcode read before it is considered unmatchable.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">2</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MAX_READS_IN_RAM_PER_TILE">--MAX_READS_IN_RAM_PER_TILE </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Configure SortingCollections to store this many records before spilling to disk. For an indexed run, each SortingCollection gets this value/number of indices. Deprecated: use `MAX_RECORDS_IN_RAM`</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">-1</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MAX_RECORDS_IN_RAM">--MAX_RECORDS_IN_RAM </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>When writing files that need to be sorted, this will specify the number of records stored in RAM before spilling to disk. Increasing this number reduces the number of file handles needed to sort the file, and increases the amount of RAM needed.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Integer</span>
			&nbsp;<span class="label">500000</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--METRICS_FILE">--METRICS_FILE </a>
			 / <small>-M</small>
		</h3>
		<p class="args">
			<b>Per-barcode and per-lane metrics written to this file.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MIN_MISMATCH_DELTA">--MIN_MISMATCH_DELTA </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Minimum difference between number of mismatches in the best and second best barcodes for a barcode to be considered a match.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">1</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MINIMUM_BASE_QUALITY">--MINIMUM_BASE_QUALITY </a>
			 / <small>-Q</small>
		</h3>
		<p class="args">
			<b>Minimum base quality. Any barcode bases falling below this quality will be considered a mismatch even if the bases match.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">0</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MINIMUM_QUALITY">--MINIMUM_QUALITY </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>The minimum quality (after transforming 0s to 1s) expected from reads.  If qualities are lower than this value, an error is thrown. The default of 2 is what the Illumina's spec describes as the minimum, but in practice the value has been observed lower.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">2</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MOLECULAR_INDEX_BASE_QUALITY_TAG">--MOLECULAR_INDEX_BASE_QUALITY_TAG </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>The tag to use to store any molecular index base qualities.  If more than one molecular index is found, their qualities will be concatenated and stored here (.i.e. the number of "M" operators in the READ_STRUCTURE)</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">QX</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MOLECULAR_INDEX_TAG">--MOLECULAR_INDEX_TAG </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>The tag to use to store any molecular indexes.  If more than one molecular index is found, they will be concatenated and stored here.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">RX</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--NUM_PROCESSORS">--NUM_PROCESSORS </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>The number of threads to run in parallel. If NUM_PROCESSORS = 0, number of cores is automatically set to the number of cores available on the machine. If NUM_PROCESSORS < 0, then the number of cores used will be the number available on the machine less NUM_PROCESSORS.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Integer</span>
			&nbsp;<span class="label">0</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--OUTPUT">--OUTPUT </a>
			 / <small>-O</small>
		</h3>
		<p class="args">
			<b>Deprecated (use LIBRARY_PARAMS).  The output SAM, BAM or CRAM file. Format is determined by extension.</b><br />
			
		</p>
			<p><b>Exclusion:</b> This argument cannot be used at the same time as <code>BARCODE_PARAMS, LIBRARY_PARAMS</code>.</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--PLATFORM">--PLATFORM </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>The name of the sequencing technology that produced the read.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">ILLUMINA</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--PROCESS_SINGLE_TILE">--PROCESS_SINGLE_TILE </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>If set, process only the tile number given and prepend the tile number to the output file name.</b><br />
			
		</p>
			<p><b>Exclusion:</b> This argument cannot be used at the same time as <code>FIRST_TILE</code>.</p>
		<p>
		<span class="label label-info ">Integer</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--QUIET">--QUIET </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Whether to suppress job-summary info on System.err.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--READ_GROUP_ID">--READ_GROUP_ID </a>
			 / <small>-RG</small>
		</h3>
		<p class="args">
			<b>ID used to link RG header record with RG tag in SAM record.  If these are unique in SAM files that get merged, merge performance is better.  If not specified, READ_GROUP_ID will be set to <first 5 chars of RUN_BARCODE>.<LANE> .</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--READ_STRUCTURE">--READ_STRUCTURE </a>
			 / <small>-RS</small>
		</h3>
		<p class="args">
			<b>A description of the logical structure of clusters in an Illumina Run, i.e. a description of the structure IlluminaBasecallsToSam assumes the  data to be in. It should consist of integer/character pairs describing the number of cycles and the type of those cycles (B for Sample Barcode, M for molecular barcode, T for Template, and S for skip).  E.g. If the input data consists of 80 base clusters and we provide a read structure of "28T8M8B8S28T" then the sequence may be split up into four reads:
* read one with 28 cycles (bases) of template
* read two with 8 cycles (bases) of molecular barcode (ex. unique molecular barcode)
* read three with 8 cycles (bases) of sample barcode
* 8 cycles (bases) skipped.
* read four with 28 cycles (bases) of template
The skipped cycles would NOT be included in an output SAM/BAM file or in read groups therein.</b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--REFERENCE_SEQUENCE">--REFERENCE_SEQUENCE </a>
			 / <small>-R</small>
		</h3>
		<p class="args">
			<b>Reference sequence file.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--RUN_BARCODE">--RUN_BARCODE </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>The barcode of the run.  Prefixed to read names.</b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--RUN_START_DATE">--RUN_START_DATE </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>The start date of the run.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Date</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--SAMPLE_ALIAS">--SAMPLE_ALIAS </a>
			 / <small>-ALIAS</small>
		</h3>
		<p class="args">
			<b>Deprecated (use LIBRARY_PARAMS).  The name of the sequenced sample</b><br />
			
		</p>
			<p><b>Exclusion:</b> This argument cannot be used at the same time as <code>BARCODE_PARAMS, LIBRARY_PARAMS</code>.</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--SEQUENCING_CENTER">--SEQUENCING_CENTER </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>The name of the sequencing center that produced the reads.  Used to set the @RG->CN header tag.</b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--showHidden">--showHidden </a>
			 / <small>-showHidden</small>
		</h3>
		<p class="args">
			<b>display hidden arguments</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--SORT">--SORT </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>If true, the output records are sorted by read name. Otherwise they are unsorted.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">true</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--TAG_PER_MOLECULAR_INDEX">--TAG_PER_MOLECULAR_INDEX </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>The list of tags to store each molecular index.  The number of tags should match the number of molecular indexes.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">List[String]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--THREE_PRIME_ADAPTER">--THREE_PRIME_ADAPTER </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>For specifying adapters other than standard Illumina</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--TILE_LIMIT">--TILE_LIMIT </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>If set, process no more than this many tiles (used for debugging).</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Integer</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--TMP_DIR">--TMP_DIR </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>One or more directories with space available to be used by this program for temporary storage of working files</b><br />
			
		</p>
		<p>
		<span class="label label-info ">List[File]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--USE_JDK_DEFLATER">--USE_JDK_DEFLATER </a>
			 / <small>-use_jdk_deflater</small>
		</h3>
		<p class="args">
			<b>Use the JDK Deflater instead of the Intel Deflater for writing compressed output</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--USE_JDK_INFLATER">--USE_JDK_INFLATER </a>
			 / <small>-use_jdk_inflater</small>
		</h3>
		<p class="args">
			<b>Use the JDK Inflater instead of the Intel Inflater for reading compressed input</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--VALIDATION_STRINGENCY">--VALIDATION_STRINGENCY </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Validation stringency for all SAM files read by this program.  Setting stringency to SILENT can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded.</b><br />
			
		</p>
			<p>
				The --VALIDATION_STRINGENCY argument is an enumerated type (ValidationStringency), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">STRICT</dt>
					<dd class="enum"></dd>
					<dt class="enum">LENIENT</dt>
					<dd class="enum"></dd>
					<dt class="enum">SILENT</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">ValidationStringency</span>
			&nbsp;<span class="label">STRICT</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--VERBOSITY">--VERBOSITY </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Control verbosity of logging.</b><br />
			
		</p>
			<p>
				The --VERBOSITY argument is an enumerated type (LogLevel), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">ERROR</dt>
					<dd class="enum"></dd>
					<dt class="enum">WARNING</dt>
					<dd class="enum"></dd>
					<dt class="enum">INFO</dt>
					<dd class="enum"></dd>
					<dt class="enum">DEBUG</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">LogLevel</span>
			&nbsp;<span class="label">INFO</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--version">--version </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>display the version number for this tool</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>

        <hr>
        <p><a href='#top'><i class='fa fa-chevron-up'></i> Return to top</a></p>
        <hr>
        <p class="see-also">See also 
        	<a href="https://www.broadinstitute.org/picard/guide/">Picard Documentation Index</a> |
        	<a href="index.html">Tool Docs Index</a> |
        	<a href="http://gatkforums.broadinstitute.org/">Support Forum</a>
        </p>

        <p class="version">Picard version 3.0.0-17-g83b016e7f-SNAPSHOT built at 27-43-2023 01:43:26.
        </p>

	</div>

	<?php printFooter($module); ?>