<?php

    include '../../../common/include/common.php';
    include_once '../../config.php';
    $module = modules::GATK;
    printHeader($module, "Picard | Tool Documentation Index", "Guide");
?>

<link type='text/css' rel='stylesheet' href='picardDoc.css'>

<div class='row-fluid' id="top">

	

	<?php $group = 'Read Data Manipulation'; ?>

	<section class="span4">
		<aside class="well">
			<a href="index.html"><h4><i class='fa fa-chevron-left'></i> Back to Tool Docs Index</h4></a>
		</aside>
		<aside class="well">
			<h2>Categories</h2>
        <style>
            #sidenav .accordion-body a {
                color : gray;
            }

            .accordion-body li {
                list-style : none;
            }
        </style>
        <ul class="nav nav-pills nav-stacked" id="sidenav">
        		<hr>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#BaseCalling">Base Calling</a>
							<div id="BaseCalling"
								<?php echo ($group == 'Base Calling')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_illumina_CheckIlluminaDirectory.html">CheckIlluminaDirectory</a>
											</li>
											<li>
												<a href="picard_illumina_CollectIlluminaBasecallingMetrics.html">CollectIlluminaBasecallingMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_CollectIlluminaLaneMetrics.html">CollectIlluminaLaneMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_ExtractIlluminaBarcodes.html">ExtractIlluminaBarcodes</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaBasecallsToFastq.html">IlluminaBasecallsToFastq</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaBasecallsToSam.html">IlluminaBasecallsToSam</a>
											</li>
											<li>
												<a href="picard_illumina_MarkIlluminaAdapters.html">MarkIlluminaAdapters</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#DiagnosticsandQualityControl">Diagnostics and Quality Control</a>
							<div id="DiagnosticsandQualityControl"
								<?php echo ($group == 'Diagnostics and Quality Control')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_AccumulateQualityYieldMetrics.html">AccumulateQualityYieldMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_AccumulateVariantCallingMetrics.html">AccumulateVariantCallingMetrics</a>
											</li>
											<li>
												<a href="picard_sam_BamIndexStats.html">BamIndexStats</a>
											</li>
											<li>
												<a href="picard_fingerprint_CalculateFingerprintMetrics.html">CalculateFingerprintMetrics</a>
											</li>
											<li>
												<a href="picard_sam_CalculateReadGroupChecksum.html">CalculateReadGroupChecksum</a>
											</li>
											<li>
												<a href="picard_fingerprint_CheckFingerprint.html">CheckFingerprint</a>
											</li>
											<li>
												<a href="picard_sam_CheckTerminatorBlock.html">CheckTerminatorBlock</a>
											</li>
											<li>
												<a href="picard_fingerprint_ClusterCrosscheckMetrics.html">ClusterCrosscheckMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectAlignmentSummaryMetrics.html">CollectAlignmentSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_CollectArraysVariantCallingMetrics.html">CollectArraysVariantCallingMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectBaseDistributionByCycle.html">CollectBaseDistributionByCycle</a>
											</li>
											<li>
												<a href="picard_analysis_CollectGcBiasMetrics.html">CollectGcBiasMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_quality_CollectHiSeqXPfFailMetrics.html">CollectHiSeqXPfFailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_directed_CollectHsMetrics.html">CollectHsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_replicates_CollectIndependentReplicateMetrics.html">CollectIndependentReplicateMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectInsertSizeMetrics.html">CollectInsertSizeMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectJumpingLibraryMetrics.html">CollectJumpingLibraryMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectMultipleMetrics.html">CollectMultipleMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectOxoGMetrics.html">CollectOxoGMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectQualityYieldMetrics.html">CollectQualityYieldMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRawWgsMetrics.html">CollectRawWgsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRnaSeqMetrics.html">CollectRnaSeqMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRrbsMetrics.html">CollectRrbsMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_CollectSamErrorMetrics.html">CollectSamErrorMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_CollectSequencingArtifactMetrics.html">CollectSequencingArtifactMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_directed_CollectTargetedPcrMetrics.html">CollectTargetedPcrMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_CollectVariantCallingMetrics.html">CollectVariantCallingMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectWgsMetrics.html">CollectWgsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectWgsMetricsWithNonZeroCoverage.html">CollectWgsMetricsWithNonZeroCoverage</a>
											</li>
											<li>
												<a href="picard_analysis_CompareMetrics.html">CompareMetrics</a>
											</li>
											<li>
												<a href="picard_sam_CompareSAMs.html">CompareSAMs</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_ConvertSequencingArtifactToOxoG.html">ConvertSequencingArtifactToOxoG</a>
											</li>
											<li>
												<a href="picard_fingerprint_CrosscheckFingerprints.html">CrosscheckFingerprints</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_EstimateLibraryComplexity.html">EstimateLibraryComplexity</a>
											</li>
											<li>
												<a href="picard_analysis_MeanQualityByCycle.html">MeanQualityByCycle</a>
											</li>
											<li>
												<a href="picard_analysis_QualityScoreDistribution.html">QualityScoreDistribution</a>
											</li>
											<li>
												<a href="picard_sam_ValidateSamFile.html">ValidateSamFile</a>
											</li>
											<li>
												<a href="picard_sam_ViewSam.html">ViewSam</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#GenotypingArraysManipulation">Genotyping Arrays Manipulation</a>
							<div id="GenotypingArraysManipulation"
								<?php echo ($group == 'Genotyping Arrays Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_arrays_illumina_BpmToNormalizationManifestCsv.html">BpmToNormalizationManifestCsv</a>
											</li>
											<li>
												<a href="picard_arrays_CombineGenotypingArrayVcfs.html">CombineGenotypingArrayVcfs</a>
											</li>
											<li>
												<a href="picard_arrays_CreateBafRegressMetricsFile.html">CreateBafRegressMetricsFile</a>
											</li>
											<li>
												<a href="picard_arrays_CreateVerifyIDIntensityContaminationMetricsFile.html">CreateVerifyIDIntensityContaminationMetricsFile</a>
											</li>
											<li>
												<a href="picard_arrays_GtcToVcf.html">GtcToVcf</a>
											</li>
											<li>
												<a href="picard_arrays_MergePedIntoVcf.html">MergePedIntoVcf</a>
											</li>
											<li>
												<a href="picard_arrays_VcfToAdpc.html">VcfToAdpc</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#IntervalsManipulation">Intervals Manipulation</a>
							<div id="IntervalsManipulation"
								<?php echo ($group == 'Intervals Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_BedToIntervalList.html">BedToIntervalList</a>
											</li>
											<li>
												<a href="picard_util_IntervalListToBed.html">IntervalListToBed</a>
											</li>
											<li>
												<a href="picard_util_IntervalListTools.html">IntervalListTools</a>
											</li>
											<li>
												<a href="picard_util_LiftOverIntervalList.html">LiftOverIntervalList</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Other">Other</a>
							<div id="Other"
								<?php echo ($group == 'Other')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_FifoBuffer.html">FifoBuffer</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReadDataManipulation">Read Data Manipulation</a>
							<div id="ReadDataManipulation"
								<?php echo ($group == 'Read Data Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_sam_AddCommentsToBam.html">AddCommentsToBam</a>
											</li>
											<li>
												<a href="picard_sam_AddOATag.html">AddOATag</a>
											</li>
											<li>
												<a href="picard_sam_AddOrReplaceReadGroups.html">AddOrReplaceReadGroups</a>
											</li>
											<li>
												<a href="picard_fastq_BamToBfq.html">BamToBfq</a>
											</li>
											<li>
												<a href="picard_sam_BuildBamIndex.html">BuildBamIndex</a>
											</li>
											<li>
												<a href="picard_sam_CleanSam.html">CleanSam</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_CollectDuplicateMetrics.html">CollectDuplicateMetrics</a>
											</li>
											<li>
												<a href="picard_sam_DownsampleSam.html">DownsampleSam</a>
											</li>
											<li>
												<a href="picard_sam_FastqToSam.html">FastqToSam</a>
											</li>
											<li>
												<a href="picard_sam_FilterSamReads.html">FilterSamReads</a>
											</li>
											<li>
												<a href="picard_sam_FixMateInformation.html">FixMateInformation</a>
											</li>
											<li>
												<a href="picard_sam_GatherBamFiles.html">GatherBamFiles</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_MarkDuplicates.html">MarkDuplicates</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_MarkDuplicatesWithMateCigar.html">MarkDuplicatesWithMateCigar</a>
											</li>
											<li>
												<a href="picard_sam_MergeBamAlignment.html">MergeBamAlignment</a>
											</li>
											<li>
												<a href="picard_sam_MergeSamFiles.html">MergeSamFiles</a>
											</li>
											<li>
												<a href="picard_sam_PositionBasedDownsampleSam.html">PositionBasedDownsampleSam</a>
											</li>
											<li>
												<a href="picard_sam_ReorderSam.html">ReorderSam</a>
											</li>
											<li>
												<a href="picard_sam_ReplaceSamHeader.html">ReplaceSamHeader</a>
											</li>
											<li>
												<a href="picard_sam_RevertOriginalBaseQualitiesAndAddMateCigar.html">RevertOriginalBaseQualitiesAndAddMateCigar</a>
											</li>
											<li>
												<a href="picard_sam_RevertSam.html">RevertSam</a>
											</li>
											<li>
												<a href="picard_sam_SamFormatConverter.html">SamFormatConverter</a>
											</li>
											<li>
												<a href="picard_sam_SamToFastq.html">SamToFastq</a>
											</li>
											<li>
												<a href="picard_sam_SetNmAndUqTags.html">SetNmAndUqTags</a>
											</li>
											<li>
												<a href="picard_sam_SetNmMdAndUqTags.html">SetNmMdAndUqTags</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_SimpleMarkDuplicatesWithMateCigar.html">SimpleMarkDuplicatesWithMateCigar</a>
											</li>
											<li>
												<a href="picard_sam_SortSam.html">SortSam</a>
											</li>
											<li>
												<a href="picard_sam_SplitSamByLibrary.html">SplitSamByLibrary</a>
											</li>
											<li>
												<a href="picard_sam_SplitSamByNumberOfReads.html">SplitSamByNumberOfReads</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_UmiAwareMarkDuplicatesWithMateCigar.html">UmiAwareMarkDuplicatesWithMateCigar</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Reference">Reference</a>
							<div id="Reference"
								<?php echo ($group == 'Reference')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_BaitDesigner.html">BaitDesigner</a>
											</li>
											<li>
												<a href="picard_sam_CreateSequenceDictionary.html">CreateSequenceDictionary</a>
											</li>
											<li>
												<a href="picard_reference_ExtractSequences.html">ExtractSequences</a>
											</li>
											<li>
												<a href="picard_reference_NonNFastaSize.html">NonNFastaSize</a>
											</li>
											<li>
												<a href="picard_reference_NormalizeFasta.html">NormalizeFasta</a>
											</li>
											<li>
												<a href="picard_util_ScatterIntervalsByNs.html">ScatterIntervalsByNs</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantEvaluationandRefinement">Variant Evaluation and Refinement</a>
							<div id="VariantEvaluationandRefinement"
								<?php echo ($group == 'Variant Evaluation and Refinement')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_vcf_MendelianViolations_FindMendelianViolations.html">FindMendelianViolations</a>
											</li>
											<li>
												<a href="picard_vcf_GenotypeConcordance.html">GenotypeConcordance</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantFiltering">Variant Filtering</a>
							<div id="VariantFiltering"
								<?php echo ($group == 'Variant Filtering')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_vcf_filter_FilterVcf.html">FilterVcf</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantManipulation">Variant Manipulation</a>
							<div id="VariantManipulation"
								<?php echo ($group == 'Variant Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_vcf_FixVcfHeader.html">FixVcfHeader</a>
											</li>
											<li>
												<a href="picard_vcf_GatherVcfs.html">GatherVcfs</a>
											</li>
											<li>
												<a href="picard_vcf_LiftoverVcf.html">LiftoverVcf</a>
											</li>
											<li>
												<a href="picard_vcf_MakeSitesOnlyVcf.html">MakeSitesOnlyVcf</a>
											</li>
											<li>
												<a href="picard_vcf_MergeVcfs.html">MergeVcfs</a>
											</li>
											<li>
												<a href="picard_vcf_RenameSampleInVcf.html">RenameSampleInVcf</a>
											</li>
											<li>
												<a href="picard_vcf_SortVcf.html">SortVcf</a>
											</li>
											<li>
												<a href="picard_vcf_SplitVcfs.html">SplitVcfs</a>
											</li>
											<li>
												<a href="picard_vcf_UpdateVcfSequenceDictionary.html">UpdateVcfSequenceDictionary</a>
											</li>
											<li>
												<a href="picard_vcf_VcfFormatConverter.html">VcfFormatConverter</a>
											</li>
											<li>
												<a href="picard_vcf_VcfToIntervalList.html">VcfToIntervalList</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Metrics">Metrics</a>
							<div id="Metrics"
								<?php echo ($group == 'Metrics')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_analysis_AlignmentSummaryMetrics.html">AlignmentSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_CollectArraysVariantCallingMetrics$ArraysControlCodesSummaryMetrics.html">ArraysControlCodesSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_CollectArraysVariantCallingMetrics$ArraysVariantCallingDetailMetrics.html">ArraysVariantCallingDetailMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_CollectArraysVariantCallingMetrics$ArraysVariantCallingSummaryMetrics.html">ArraysVariantCallingSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_BafRegressMetrics.html">BafRegressMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_SequencingArtifactMetrics$BaitBiasDetailMetrics.html">BaitBiasDetailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_SequencingArtifactMetrics$BaitBiasSummaryMetrics.html">BaitBiasSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_BarcodeMetric.html">BarcodeMetric</a>
											</li>
											<li>
												<a href="picard_analysis_BaseDistributionByCycleMetrics.html">BaseDistributionByCycleMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_BaseErrorMetric.html">BaseErrorMetric</a>
											</li>
											<li>
												<a href="picard_fingerprint_ClusteredCrosscheckMetric.html">ClusteredCrosscheckMetric</a>
											</li>
											<li>
												<a href="picard_analysis_CollectOxoGMetrics$CpcgMetrics.html">CpcgMetrics</a>
											</li>
											<li>
												<a href="picard_sam_DuplicationMetrics.html">DuplicationMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_ErrorMetric.html">ErrorMetric</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_ErrorSummaryMetrics.html">ErrorSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_fingerprint_FingerprintMetrics.html">FingerprintMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_FingerprintingDetailMetrics.html">FingerprintingDetailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_FingerprintingSummaryMetrics.html">FingerprintingSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_GcBiasDetailMetrics.html">GcBiasDetailMetrics</a>
											</li>
											<li>
												<a href="picard_metrics_GcBiasMetrics.html">GcBiasMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_GcBiasSummaryMetrics.html">GcBiasSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_GenotypeConcordanceContingencyMetrics.html">GenotypeConcordanceContingencyMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_GenotypeConcordanceDetailMetrics.html">GenotypeConcordanceDetailMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_GenotypeConcordanceSummaryMetrics.html">GenotypeConcordanceSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_directed_HsMetrics.html">HsMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaBasecallingMetrics.html">IlluminaBasecallingMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaLaneMetrics.html">IlluminaLaneMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaPhasingMetrics.html">IlluminaPhasingMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_IndelErrorMetric.html">IndelErrorMetric</a>
											</li>
											<li>
												<a href="picard_analysis_replicates_IndependentReplicateMetric.html">IndependentReplicateMetric</a>
											</li>
											<li>
												<a href="picard_analysis_InsertSizeMetrics.html">InsertSizeMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_JumpingLibraryMetrics.html">JumpingLibraryMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_MendelianViolations_MendelianViolationMetrics.html">MendelianViolationMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_OverlappingErrorMetric.html">OverlappingErrorMetric</a>
											</li>
											<li>
												<a href="picard_illumina_quality_CollectHiSeqXPfFailMetrics$PFFailDetailedMetric.html">PFFailDetailedMetric</a>
											</li>
											<li>
												<a href="picard_illumina_quality_CollectHiSeqXPfFailMetrics$PFFailSummaryMetric.html">PFFailSummaryMetric</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_SequencingArtifactMetrics$PreAdapterDetailMetrics.html">PreAdapterDetailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_SequencingArtifactMetrics$PreAdapterSummaryMetrics.html">PreAdapterSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectQualityYieldMetrics$QualityYieldMetrics.html">QualityYieldMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRawWgsMetrics$RawWgsMetrics.html">RawWgsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_RnaSeqMetrics.html">RnaSeqMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_RrbsCpgDetailMetrics.html">RrbsCpgDetailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_RrbsSummaryMetrics.html">RrbsSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamComparisonMetric.html">SamComparisonMetric</a>
											</li>
											<li>
												<a href="picard_analysis_directed_TargetMetrics.html">TargetMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_directed_TargetedPcrMetrics.html">TargetedPcrMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_TheoreticalSensitivityMetrics.html">TheoreticalSensitivityMetrics</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_UmiMetrics.html">UmiMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_CollectVariantCallingMetrics$VariantCallingDetailMetrics.html">VariantCallingDetailMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_CollectVariantCallingMetrics$VariantCallingSummaryMetrics.html">VariantCallingSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_VerifyIDIntensityContaminationMetrics.html">VerifyIDIntensityContaminationMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_WgsMetrics.html">WgsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectWgsMetricsWithNonZeroCoverage$WgsMetricsWithNonZeroCoverage.html">WgsMetricsWithNonZeroCoverage</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
        </ul>
		</aside>
		<?php getForumPosts( 'SimpleMarkDuplicatesWithMateCigar' ) ?>

	</section>

	<div class="span8">

		<h1>SimpleMarkDuplicatesWithMateCigar</h1>

		<p class="lead">Examines aligned records in the supplied SAM/BAM/CRAM file to locate duplicate molecules. All records are then written to the output file with the duplicate records flagged.</p>

			<h3>Category
				<small> Read Data Manipulation</small>
			</h3>
		<hr>
		<h2>Overview</h2>
		This is a simple tool to mark duplicates using the DuplicateSetIterator, DuplicateSet, and SAMRecordDuplicateComparator.
 
 Users should continue to use MarkDuplicates in general.  The main motivation of this tool was the fact that 
 MarkDuplicates has many, many, many useful test cases, but few unit tests for validating individual duplicate sets. To
 test the DuplicateSetIterator, DuplicateSet, and SAMRecordDuplicateComparator, the most expedient method was to write
 this tool and make sure it behaves similarly to MarkDuplicates.  Not the best, I know, but good enough.  NH 06/25/2015.
  
 
 See MarkDuplicates for more details.
@author nhomer


				<h3>SimpleMarkDuplicatesWithMateCigar specific arguments</h3>
				<p>This table summarizes the command-line arguments that are specific to this tool. For more details on each argument, see the list further down below the table or click on an argument name to jump directly to that entry in the list.</p>
				<table class="table table-striped table-bordered table-condensed">
					<thead>
					<tr>
						<th>Argument name(s)</th>
						<th>Default value</th>
						<th>Summary</th>
					</tr>
					</thead>
					<tbody>
			<tr>
				<th colspan="4" id="row-divider">Required Arguments</th>
			</tr>
				<tr>
					<td><a href="#--INPUT">--INPUT</a><br />
								&nbsp;<em>-I</em>
					</td>
					<!--<td>List[String]</td> -->
					<td>[]</td>
					<td>One or more input SAM, BAM or CRAM files to analyze. Must be coordinate sorted.</td>
				</tr>
				<tr>
					<td><a href="#--METRICS_FILE">--METRICS_FILE</a><br />
								&nbsp;<em>-M</em>
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>File to write duplication metrics to</td>
				</tr>
				<tr>
					<td><a href="#--OUTPUT">--OUTPUT</a><br />
								&nbsp;<em>-O</em>
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>The output file to write marked records to</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Tool Arguments</th>
			</tr>
				<tr>
					<td><a href="#--arguments_file">--arguments_file</a><br />
					</td>
					<!--<td>List[File]</td> -->
					<td>[]</td>
					<td>read one or more arguments files and add them to the command line</td>
				</tr>
				<tr>
					<td><a href="#--ASSUME_SORT_ORDER">--ASSUME_SORT_ORDER</a><br />
								&nbsp;<em>-ASO</em>
					</td>
					<!--<td>SortOrder</td> -->
					<td>null</td>
					<td>If not null, assume that the input file has this order even if the header says otherwise.</td>
				</tr>
				<tr>
					<td><a href="#--ASSUME_SORTED">--ASSUME_SORTED</a><br />
								&nbsp;<em>-AS</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>If true, assume that the input file is coordinate sorted even if the header says otherwise. Deprecated, used ASSUME_SORT_ORDER=coordinate instead.</td>
				</tr>
				<tr>
					<td><a href="#--BARCODE_TAG">--BARCODE_TAG</a><br />
					</td>
					<!--<td>String</td> -->
					<td>null</td>
					<td>Barcode SAM tag (ex. BC for 10X Genomics)</td>
				</tr>
				<tr>
					<td><a href="#--CLEAR_DT">--CLEAR_DT</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>true</td>
					<td>Clear DT tag from input SAM records. Should be set to false if input SAM doesn't have this tag.  Default true</td>
				</tr>
				<tr>
					<td><a href="#--COMMENT">--COMMENT</a><br />
								&nbsp;<em>-CO</em>
					</td>
					<!--<td>List[String]</td> -->
					<td>[]</td>
					<td>Comment(s) to include in the output file's header.</td>
				</tr>
				<tr>
					<td><a href="#--DUPLEX_UMI">--DUPLEX_UMI</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Treat UMIs as being duplex stranded.  This option requires that the UMI consist of two equal length strings that are separated by a hyphen (e.g. 'ATC-GTC'). Reads are considered duplicates if, in addition to standard definition, have identical normalized UMIs.  A UMI from the 'bottom' strand is normalized by swapping its content around the hyphen (eg. ATC-GTC becomes GTC-ATC).  A UMI from the 'top' strand is already normalized as it is. Both reads from a read pair considered top strand if the read 1 unclipped 5' coordinate is less than the read 2 unclipped 5' coordinate. All chimeric reads and read fragments are treated as having come from the top strand. With this option is it required that the BARCODE_TAG hold non-normalized UMIs. Default false.</td>
				</tr>
				<tr>
					<td><a href="#--DUPLICATE_SCORING_STRATEGY">--DUPLICATE_SCORING_STRATEGY</a><br />
								&nbsp;<em>-DS</em>
					</td>
					<!--<td>ScoringStrategy</td> -->
					<td>SUM_OF_BASE_QUALITIES</td>
					<td>The scoring strategy for choosing the non-duplicate among candidates.</td>
				</tr>
				<tr>
					<td><a href="#--FLOW_EFFECTIVE_QUALITY_THRESHOLD">--FLOW_EFFECTIVE_QUALITY_THRESHOLD</a><br />
					</td>
					<!--<td>int</td> -->
					<td>15</td>
					<td>Threshold for considering a quality value high enough to be included when calculating FLOW_QUALITY_SUM_STRATEGY calculation.</td>
				</tr>
				<tr>
					<td><a href="#--FLOW_MODE">--FLOW_MODE</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>enable parameters and behavior specific to flow based reads.</td>
				</tr>
				<tr>
					<td><a href="#--FLOW_Q_IS_KNOWN_END">--FLOW_Q_IS_KNOWN_END</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Treat position of read trimming based on quality as the known end (relevant for flow based reads). Default false - if the read is trimmed on quality its end is not defined and the read is duplicate of any read starting at the same place.</td>
				</tr>
				<tr>
					<td><a href="#--FLOW_QUALITY_SUM_STRATEGY">--FLOW_QUALITY_SUM_STRATEGY</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Use specific quality summing strategy for flow based reads. The strategy ensures that the same (and correct) quality value is used for all bases of the same homopolymer.</td>
				</tr>
				<tr>
					<td><a href="#--FLOW_SKIP_FIRST_N_FLOWS">--FLOW_SKIP_FIRST_N_FLOWS</a><br />
					</td>
					<!--<td>int</td> -->
					<td>0</td>
					<td>Skip first N flows, starting from the read's start, when considering duplicates. Useful for flow based reads where sometimes there is noise in the first flows (for this argument, "read start" means 5' end).</td>
				</tr>
				<tr>
					<td><a href="#--help">--help</a><br />
								&nbsp;<em>-h</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>display the help message</td>
				</tr>
				<tr>
					<td><a href="#--MAX_FILE_HANDLES_FOR_READ_ENDS_MAP">--MAX_FILE_HANDLES_FOR_READ_ENDS_MAP</a><br />
								&nbsp;<em>-MAX_FILE_HANDLES</em>
					</td>
					<!--<td>int</td> -->
					<td>8000</td>
					<td>Maximum number of file handles to keep open when spilling read ends to disk. Set this number a little lower than the per-process maximum number of file that may be open. This number can be found by executing the 'ulimit -n' command on a Unix system.</td>
				</tr>
				<tr>
					<td><a href="#--MAX_OPTICAL_DUPLICATE_SET_SIZE">--MAX_OPTICAL_DUPLICATE_SET_SIZE</a><br />
					</td>
					<!--<td>long</td> -->
					<td>300000</td>
					<td>This number is the maximum size of a set of duplicate reads for which we will attempt to determine which are optical duplicates.  Please be aware that if you raise this value too high and do encounter a very large set of duplicate reads, it will severely affect the runtime of this tool.  To completely disable this check, set the value to -1.</td>
				</tr>
				<tr>
					<td><a href="#--MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP">--MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP</a><br />
								&nbsp;<em>-MAX_SEQS</em>
					</td>
					<!--<td>int</td> -->
					<td>50000</td>
					<td>This option is obsolete. ReadEnds will always be spilled to disk.</td>
				</tr>
				<tr>
					<td><a href="#--MOLECULAR_IDENTIFIER_TAG">--MOLECULAR_IDENTIFIER_TAG</a><br />
					</td>
					<!--<td>String</td> -->
					<td>null</td>
					<td>SAM tag to uniquely identify the molecule from which a read was derived.  Use of this option requires that the BARCODE_TAG option be set to a non null value.  Default null.</td>
				</tr>
				<tr>
					<td><a href="#--OPTICAL_DUPLICATE_PIXEL_DISTANCE">--OPTICAL_DUPLICATE_PIXEL_DISTANCE</a><br />
					</td>
					<!--<td>int</td> -->
					<td>100</td>
					<td>The maximum offset between two duplicate clusters in order to consider them optical duplicates. The default is appropriate for unpatterned versions of the Illumina platform. For the patterned flowcell models, 2500 is moreappropriate. For other platforms and models, users should experiment to find what works best.</td>
				</tr>
				<tr>
					<td><a href="#--PROGRAM_GROUP_COMMAND_LINE">--PROGRAM_GROUP_COMMAND_LINE</a><br />
								&nbsp;<em>-PG_COMMAND</em>
					</td>
					<!--<td>String</td> -->
					<td>null</td>
					<td>Value of CL tag of PG record to be created. If not supplied the command line will be detected automatically.</td>
				</tr>
				<tr>
					<td><a href="#--PROGRAM_GROUP_NAME">--PROGRAM_GROUP_NAME</a><br />
								&nbsp;<em>-PG_NAME</em>
					</td>
					<!--<td>String</td> -->
					<td>SimpleMarkDuplicatesWithMateCigar</td>
					<td>Value of PN tag of PG record to be created.</td>
				</tr>
				<tr>
					<td><a href="#--PROGRAM_GROUP_VERSION">--PROGRAM_GROUP_VERSION</a><br />
								&nbsp;<em>-PG_VERSION</em>
					</td>
					<!--<td>String</td> -->
					<td>null</td>
					<td>Value of VN tag of PG record to be created. If not specified, the version will be detected automatically.</td>
				</tr>
				<tr>
					<td><a href="#--PROGRAM_RECORD_ID">--PROGRAM_RECORD_ID</a><br />
								&nbsp;<em>-PG</em>
					</td>
					<!--<td>String</td> -->
					<td>MarkDuplicates</td>
					<td>The program record ID for the @PG record(s) created by this program. Set to null to disable PG record creation.  This string may have a suffix appended to avoid collision with other program record IDs.</td>
				</tr>
				<tr>
					<td><a href="#--READ_NAME_REGEX">--READ_NAME_REGEX</a><br />
					</td>
					<!--<td>String</td> -->
					<td><optimized capture of last three ':' separated fields as numeric values></td>
					<td>MarkDuplicates can use the tile and cluster positions to estimate the rate of optical duplication in addition to the dominant source of duplication, PCR, to provide a more accurate estimation of library size. By default (with no READ_NAME_REGEX specified), MarkDuplicates will attempt to extract coordinates using a split on ':' (see Note below).  Set READ_NAME_REGEX to 'null' to disable optical duplicate detection. Note that without optical duplicate counts, library size estimation will be less accurate. If the read name does not follow a standard Illumina colon-separation convention, but does contain tile and x,y coordinates, a regular expression can be specified to extract three variables: tile/region, x coordinate and y coordinate from a read name. The regular expression must contain three capture groups for the three variables, in order. It must match the entire read name.   e.g. if field names were separated by semi-colon (';') this example regex could be specified      (?:.*;)?([0-9]+)[^;]*;([0-9]+)[^;]*;([0-9]+)[^;]*$ Note that if no READ_NAME_REGEX is specified, the read name is split on ':'.   For 5 element names, the 3rd, 4th and 5th elements are assumed to be tile, x and y values.   For 7 element names (CASAVA 1.8), the 5th, 6th, and 7th elements are assumed to be tile, x and y values.</td>
				</tr>
				<tr>
					<td><a href="#--READ_ONE_BARCODE_TAG">--READ_ONE_BARCODE_TAG</a><br />
					</td>
					<!--<td>String</td> -->
					<td>null</td>
					<td>Read one barcode SAM tag (ex. BX for 10X Genomics)</td>
				</tr>
				<tr>
					<td><a href="#--READ_TWO_BARCODE_TAG">--READ_TWO_BARCODE_TAG</a><br />
					</td>
					<!--<td>String</td> -->
					<td>null</td>
					<td>Read two barcode SAM tag (ex. BX for 10X Genomics)</td>
				</tr>
				<tr>
					<td><a href="#--REMOVE_DUPLICATES">--REMOVE_DUPLICATES</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>If true do not write duplicates to the output file instead of writing them with appropriate flags set.</td>
				</tr>
				<tr>
					<td><a href="#--REMOVE_SEQUENCING_DUPLICATES">--REMOVE_SEQUENCING_DUPLICATES</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>If true remove 'optical' duplicates and other duplicates that appear to have arisen from the sequencing process instead of the library preparation process, even if REMOVE_DUPLICATES is false. If REMOVE_DUPLICATES is true, all duplicates are removed and this option is ignored.</td>
				</tr>
				<tr>
					<td><a href="#--SORTING_COLLECTION_SIZE_RATIO">--SORTING_COLLECTION_SIZE_RATIO</a><br />
					</td>
					<!--<td>double</td> -->
					<td>0.25</td>
					<td>This number, plus the maximum RAM available to the JVM, determine the memory footprint used by some of the sorting collections.  If you are running out of memory, try reducing this number.</td>
				</tr>
				<tr>
					<td><a href="#--TAG_DUPLICATE_SET_MEMBERS">--TAG_DUPLICATE_SET_MEMBERS</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>If a read appears in a duplicate set, add two tags. The first tag, DUPLICATE_SET_SIZE_TAG (DS), indicates the size of the duplicate set. The smallest possible DS value is 2 which occurs when two reads map to the same portion of the reference only one of which is marked as duplicate. The second tag, DUPLICATE_SET_INDEX_TAG (DI), represents a unique identifier for the duplicate set to which the record belongs. This identifier is the index-in-file of the representative read that was selected out of the duplicate set.</td>
				</tr>
				<tr>
					<td><a href="#--TAGGING_POLICY">--TAGGING_POLICY</a><br />
					</td>
					<!--<td>DuplicateTaggingPolicy</td> -->
					<td>DontTag</td>
					<td>Determines how duplicate types are recorded in the DT optional attribute.</td>
				</tr>
				<tr>
					<td><a href="#--UNPAIRED_END_UNCERTAINTY">--UNPAIRED_END_UNCERTAINTY</a><br />
					</td>
					<!--<td>int</td> -->
					<td>0</td>
					<td>Maximal difference of the read end position that counted as equal. Useful for flow based reads where the end position might vary due to sequencing errors. (for this argument, "read end" means 3' end)</td>
				</tr>
				<tr>
					<td><a href="#--USE_END_IN_UNPAIRED_READS">--USE_END_IN_UNPAIRED_READS</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Make the end location of single end read be significant when considering duplicates, in addition to the start location, which is always significant (i.e. require single-ended reads to start andend on the same position to be considered duplicate) (for this argument, "read end" means 3' end).</td>
				</tr>
				<tr>
					<td><a href="#--USE_UNPAIRED_CLIPPED_END">--USE_UNPAIRED_CLIPPED_END</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Use position of the clipping as the end position, when considering duplicates (or use the unclipped end position) (for this argument, "read end" means 3' end).</td>
				</tr>
				<tr>
					<td><a href="#--version">--version</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>display the version number for this tool</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Common Arguments</th>
			</tr>
				<tr>
					<td><a href="#--ADD_PG_TAG_TO_READS">--ADD_PG_TAG_TO_READS</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>true</td>
					<td>Add PG tag to each read in a SAM or BAM</td>
				</tr>
				<tr>
					<td><a href="#--COMPRESSION_LEVEL">--COMPRESSION_LEVEL</a><br />
					</td>
					<!--<td>int</td> -->
					<td>5</td>
					<td>Compression level for all compressed files created (e.g. BAM and VCF).</td>
				</tr>
				<tr>
					<td><a href="#--CREATE_INDEX">--CREATE_INDEX</a><br />
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Whether to create an index when writing VCF or coordinate sorted BAM output.</td>
				</tr>
				<tr>
					<td><a href="#--CREATE_MD5_FILE">--CREATE_MD5_FILE</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Whether to create an MD5 digest for any BAM or FASTQ files created.  </td>
				</tr>
				<tr>
					<td><a href="#--MAX_RECORDS_IN_RAM">--MAX_RECORDS_IN_RAM</a><br />
					</td>
					<!--<td>Integer</td> -->
					<td>500000</td>
					<td>When writing files that need to be sorted, this will specify the number of records stored in RAM before spilling to disk. Increasing this number reduces the number of file handles needed to sort the file, and increases the amount of RAM needed.</td>
				</tr>
				<tr>
					<td><a href="#--QUIET">--QUIET</a><br />
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Whether to suppress job-summary info on System.err.</td>
				</tr>
				<tr>
					<td><a href="#--REFERENCE_SEQUENCE">--REFERENCE_SEQUENCE</a><br />
								&nbsp;<em>-R</em>
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>Reference sequence file.</td>
				</tr>
				<tr>
					<td><a href="#--TMP_DIR">--TMP_DIR</a><br />
					</td>
					<!--<td>List[File]</td> -->
					<td>[]</td>
					<td>One or more directories with space available to be used by this program for temporary storage of working files</td>
				</tr>
				<tr>
					<td><a href="#--USE_JDK_DEFLATER">--USE_JDK_DEFLATER</a><br />
								&nbsp;<em>-use_jdk_deflater</em>
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Use the JDK Deflater instead of the Intel Deflater for writing compressed output</td>
				</tr>
				<tr>
					<td><a href="#--USE_JDK_INFLATER">--USE_JDK_INFLATER</a><br />
								&nbsp;<em>-use_jdk_inflater</em>
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Use the JDK Inflater instead of the Intel Inflater for reading compressed input</td>
				</tr>
				<tr>
					<td><a href="#--VALIDATION_STRINGENCY">--VALIDATION_STRINGENCY</a><br />
					</td>
					<!--<td>ValidationStringency</td> -->
					<td>STRICT</td>
					<td>Validation stringency for all SAM files read by this program.  Setting stringency to SILENT can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded.</td>
				</tr>
				<tr>
					<td><a href="#--VERBOSITY">--VERBOSITY</a><br />
					</td>
					<!--<td>LogLevel</td> -->
					<td>INFO</td>
					<td>Control verbosity of logging.</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Advanced Arguments</th>
			</tr>
				<tr>
					<td><a href="#--showHidden">--showHidden</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>display hidden arguments</td>
				</tr>
					</tbody>
				</table>

					<h3>Argument details</h3>
					<p>Arguments in this list are specific to this tool. Keep in mind that other arguments are available that are shared with other tools (e.g. command-line GATK arguments); see Inherited arguments above.</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--ADD_PG_TAG_TO_READS">--ADD_PG_TAG_TO_READS </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Add PG tag to each read in a SAM or BAM</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">true</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--arguments_file">--arguments_file </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>read one or more arguments files and add them to the command line</b><br />
			
		</p>
		<p>
		<span class="label label-info ">List[File]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--ASSUME_SORT_ORDER">--ASSUME_SORT_ORDER </a>
			 / <small>-ASO</small>
		</h3>
		<p class="args">
			<b>If not null, assume that the input file has this order even if the header says otherwise.</b><br />
			
		</p>
			<p><b>Exclusion:</b> This argument cannot be used at the same time as <code>ASSUME_SORTED</code>.</p>
			<p>
				The --ASSUME_SORT_ORDER argument is an enumerated type (SortOrder), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">unsorted</dt>
					<dd class="enum"></dd>
					<dt class="enum">queryname</dt>
					<dd class="enum"></dd>
					<dt class="enum">coordinate</dt>
					<dd class="enum"></dd>
					<dt class="enum">duplicate</dt>
					<dd class="enum"></dd>
					<dt class="enum">unknown</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">SortOrder</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--ASSUME_SORTED">--ASSUME_SORTED </a>
			 / <small>-AS</small>
		</h3>
		<p class="args">
			<b>If true, assume that the input file is coordinate sorted even if the header says otherwise. Deprecated, used ASSUME_SORT_ORDER=coordinate instead.</b><br />
			
		</p>
			<p><b>Exclusion:</b> This argument cannot be used at the same time as <code>ASSUME_SORT_ORDER</code>.</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--BARCODE_TAG">--BARCODE_TAG </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Barcode SAM tag (ex. BC for 10X Genomics)</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--CLEAR_DT">--CLEAR_DT </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Clear DT tag from input SAM records. Should be set to false if input SAM doesn't have this tag.  Default true</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">true</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--COMMENT">--COMMENT </a>
			 / <small>-CO</small>
		</h3>
		<p class="args">
			<b>Comment(s) to include in the output file's header.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">List[String]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--COMPRESSION_LEVEL">--COMPRESSION_LEVEL </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Compression level for all compressed files created (e.g. BAM and VCF).</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">5</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--CREATE_INDEX">--CREATE_INDEX </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Whether to create an index when writing VCF or coordinate sorted BAM output.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--CREATE_MD5_FILE">--CREATE_MD5_FILE </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Whether to create an MD5 digest for any BAM or FASTQ files created.  </b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--DUPLEX_UMI">--DUPLEX_UMI </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Treat UMIs as being duplex stranded.  This option requires that the UMI consist of two equal length strings that are separated by a hyphen (e.g. 'ATC-GTC'). Reads are considered duplicates if, in addition to standard definition, have identical normalized UMIs.  A UMI from the 'bottom' strand is normalized by swapping its content around the hyphen (eg. ATC-GTC becomes GTC-ATC).  A UMI from the 'top' strand is already normalized as it is. Both reads from a read pair considered top strand if the read 1 unclipped 5' coordinate is less than the read 2 unclipped 5' coordinate. All chimeric reads and read fragments are treated as having come from the top strand. With this option is it required that the BARCODE_TAG hold non-normalized UMIs. Default false.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--DUPLICATE_SCORING_STRATEGY">--DUPLICATE_SCORING_STRATEGY </a>
			 / <small>-DS</small>
		</h3>
		<p class="args">
			<b>The scoring strategy for choosing the non-duplicate among candidates.</b><br />
			
		</p>
			<p>
				The --DUPLICATE_SCORING_STRATEGY argument is an enumerated type (ScoringStrategy), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">SUM_OF_BASE_QUALITIES</dt>
					<dd class="enum"></dd>
					<dt class="enum">TOTAL_MAPPED_REFERENCE_LENGTH</dt>
					<dd class="enum"></dd>
					<dt class="enum">RANDOM</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">ScoringStrategy</span>
			&nbsp;<span class="label">SUM_OF_BASE_QUALITIES</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--FLOW_EFFECTIVE_QUALITY_THRESHOLD">--FLOW_EFFECTIVE_QUALITY_THRESHOLD </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Threshold for considering a quality value high enough to be included when calculating FLOW_QUALITY_SUM_STRATEGY calculation.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">15</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--FLOW_MODE">--FLOW_MODE </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>enable parameters and behavior specific to flow based reads.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--FLOW_Q_IS_KNOWN_END">--FLOW_Q_IS_KNOWN_END </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Treat position of read trimming based on quality as the known end (relevant for flow based reads). Default false - if the read is trimmed on quality its end is not defined and the read is duplicate of any read starting at the same place.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--FLOW_QUALITY_SUM_STRATEGY">--FLOW_QUALITY_SUM_STRATEGY </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Use specific quality summing strategy for flow based reads. The strategy ensures that the same (and correct) quality value is used for all bases of the same homopolymer.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--FLOW_SKIP_FIRST_N_FLOWS">--FLOW_SKIP_FIRST_N_FLOWS </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Skip first N flows, starting from the read's start, when considering duplicates. Useful for flow based reads where sometimes there is noise in the first flows (for this argument, "read start" means 5' end).</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">0</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--help">--help </a>
			 / <small>-h</small>
		</h3>
		<p class="args">
			<b>display the help message</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--INPUT">--INPUT </a>
			 / <small>-I</small>
		</h3>
		<p class="args">
			<b>One or more input SAM, BAM or CRAM files to analyze. Must be coordinate sorted.</b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">List[String]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MAX_FILE_HANDLES_FOR_READ_ENDS_MAP">--MAX_FILE_HANDLES_FOR_READ_ENDS_MAP </a>
			 / <small>-MAX_FILE_HANDLES</small>
		</h3>
		<p class="args">
			<b>Maximum number of file handles to keep open when spilling read ends to disk. Set this number a little lower than the per-process maximum number of file that may be open. This number can be found by executing the 'ulimit -n' command on a Unix system.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">8000</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MAX_OPTICAL_DUPLICATE_SET_SIZE">--MAX_OPTICAL_DUPLICATE_SET_SIZE </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>This number is the maximum size of a set of duplicate reads for which we will attempt to determine which are optical duplicates.  Please be aware that if you raise this value too high and do encounter a very large set of duplicate reads, it will severely affect the runtime of this tool.  To completely disable this check, set the value to -1.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">long</span>
			&nbsp;<span class="label">300000</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MAX_RECORDS_IN_RAM">--MAX_RECORDS_IN_RAM </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>When writing files that need to be sorted, this will specify the number of records stored in RAM before spilling to disk. Increasing this number reduces the number of file handles needed to sort the file, and increases the amount of RAM needed.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Integer</span>
			&nbsp;<span class="label">500000</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP">--MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP </a>
			 / <small>-MAX_SEQS</small>
		</h3>
		<p class="args">
			<b>This option is obsolete. ReadEnds will always be spilled to disk.</b><br />
			If more than this many sequences in SAM file, don't spill to disk because there will not
 be enough file handles.
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">50000</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--METRICS_FILE">--METRICS_FILE </a>
			 / <small>-M</small>
		</h3>
		<p class="args">
			<b>File to write duplication metrics to</b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MOLECULAR_IDENTIFIER_TAG">--MOLECULAR_IDENTIFIER_TAG </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>SAM tag to uniquely identify the molecule from which a read was derived.  Use of this option requires that the BARCODE_TAG option be set to a non null value.  Default null.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--OPTICAL_DUPLICATE_PIXEL_DISTANCE">--OPTICAL_DUPLICATE_PIXEL_DISTANCE </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>The maximum offset between two duplicate clusters in order to consider them optical duplicates. The default is appropriate for unpatterned versions of the Illumina platform. For the patterned flowcell models, 2500 is moreappropriate. For other platforms and models, users should experiment to find what works best.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">100</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--OUTPUT">--OUTPUT </a>
			 / <small>-O</small>
		</h3>
		<p class="args">
			<b>The output file to write marked records to</b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--PROGRAM_GROUP_COMMAND_LINE">--PROGRAM_GROUP_COMMAND_LINE </a>
			 / <small>-PG_COMMAND</small>
		</h3>
		<p class="args">
			<b>Value of CL tag of PG record to be created. If not supplied the command line will be detected automatically.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--PROGRAM_GROUP_NAME">--PROGRAM_GROUP_NAME </a>
			 / <small>-PG_NAME</small>
		</h3>
		<p class="args">
			<b>Value of PN tag of PG record to be created.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">SimpleMarkDuplicatesWithMateCigar</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--PROGRAM_GROUP_VERSION">--PROGRAM_GROUP_VERSION </a>
			 / <small>-PG_VERSION</small>
		</h3>
		<p class="args">
			<b>Value of VN tag of PG record to be created. If not specified, the version will be detected automatically.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--PROGRAM_RECORD_ID">--PROGRAM_RECORD_ID </a>
			 / <small>-PG</small>
		</h3>
		<p class="args">
			<b>The program record ID for the @PG record(s) created by this program. Set to null to disable PG record creation.  This string may have a suffix appended to avoid collision with other program record IDs.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">MarkDuplicates</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--QUIET">--QUIET </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Whether to suppress job-summary info on System.err.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--READ_NAME_REGEX">--READ_NAME_REGEX </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>MarkDuplicates can use the tile and cluster positions to estimate the rate of optical duplication in addition to the dominant source of duplication, PCR, to provide a more accurate estimation of library size. By default (with no READ_NAME_REGEX specified), MarkDuplicates will attempt to extract coordinates using a split on ':' (see Note below).  Set READ_NAME_REGEX to 'null' to disable optical duplicate detection. Note that without optical duplicate counts, library size estimation will be less accurate. If the read name does not follow a standard Illumina colon-separation convention, but does contain tile and x,y coordinates, a regular expression can be specified to extract three variables: tile/region, x coordinate and y coordinate from a read name. The regular expression must contain three capture groups for the three variables, in order. It must match the entire read name.   e.g. if field names were separated by semi-colon (';') this example regex could be specified      (?:.*;)?([0-9]+)[^;]*;([0-9]+)[^;]*;([0-9]+)[^;]*$ Note that if no READ_NAME_REGEX is specified, the read name is split on ':'.   For 5 element names, the 3rd, 4th and 5th elements are assumed to be tile, x and y values.   For 7 element names (CASAVA 1.8), the 5th, 6th, and 7th elements are assumed to be tile, x and y values.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label"><optimized capture of last three ':' separated fields as numeric values></span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--READ_ONE_BARCODE_TAG">--READ_ONE_BARCODE_TAG </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Read one barcode SAM tag (ex. BX for 10X Genomics)</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--READ_TWO_BARCODE_TAG">--READ_TWO_BARCODE_TAG </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Read two barcode SAM tag (ex. BX for 10X Genomics)</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--REFERENCE_SEQUENCE">--REFERENCE_SEQUENCE </a>
			 / <small>-R</small>
		</h3>
		<p class="args">
			<b>Reference sequence file.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--REMOVE_DUPLICATES">--REMOVE_DUPLICATES </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>If true do not write duplicates to the output file instead of writing them with appropriate flags set.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--REMOVE_SEQUENCING_DUPLICATES">--REMOVE_SEQUENCING_DUPLICATES </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>If true remove 'optical' duplicates and other duplicates that appear to have arisen from the sequencing process instead of the library preparation process, even if REMOVE_DUPLICATES is false. If REMOVE_DUPLICATES is true, all duplicates are removed and this option is ignored.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--showHidden">--showHidden </a>
			 / <small>-showHidden</small>
		</h3>
		<p class="args">
			<b>display hidden arguments</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--SORTING_COLLECTION_SIZE_RATIO">--SORTING_COLLECTION_SIZE_RATIO </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>This number, plus the maximum RAM available to the JVM, determine the memory footprint used by some of the sorting collections.  If you are running out of memory, try reducing this number.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">double</span>
			&nbsp;<span class="label">0.25</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--TAG_DUPLICATE_SET_MEMBERS">--TAG_DUPLICATE_SET_MEMBERS </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>If a read appears in a duplicate set, add two tags. The first tag, DUPLICATE_SET_SIZE_TAG (DS), indicates the size of the duplicate set. The smallest possible DS value is 2 which occurs when two reads map to the same portion of the reference only one of which is marked as duplicate. The second tag, DUPLICATE_SET_INDEX_TAG (DI), represents a unique identifier for the duplicate set to which the record belongs. This identifier is the index-in-file of the representative read that was selected out of the duplicate set.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--TAGGING_POLICY">--TAGGING_POLICY </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Determines how duplicate types are recorded in the DT optional attribute.</b><br />
			
		</p>
			<p>
				The --TAGGING_POLICY argument is an enumerated type (DuplicateTaggingPolicy), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">DontTag</dt>
					<dd class="enum"></dd>
					<dt class="enum">OpticalOnly</dt>
					<dd class="enum"></dd>
					<dt class="enum">All</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">DuplicateTaggingPolicy</span>
			&nbsp;<span class="label">DontTag</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--TMP_DIR">--TMP_DIR </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>One or more directories with space available to be used by this program for temporary storage of working files</b><br />
			
		</p>
		<p>
		<span class="label label-info ">List[File]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--UNPAIRED_END_UNCERTAINTY">--UNPAIRED_END_UNCERTAINTY </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Maximal difference of the read end position that counted as equal. Useful for flow based reads where the end position might vary due to sequencing errors. (for this argument, "read end" means 3' end)</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">0</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--USE_END_IN_UNPAIRED_READS">--USE_END_IN_UNPAIRED_READS </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Make the end location of single end read be significant when considering duplicates, in addition to the start location, which is always significant (i.e. require single-ended reads to start andend on the same position to be considered duplicate) (for this argument, "read end" means 3' end).</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--USE_JDK_DEFLATER">--USE_JDK_DEFLATER </a>
			 / <small>-use_jdk_deflater</small>
		</h3>
		<p class="args">
			<b>Use the JDK Deflater instead of the Intel Deflater for writing compressed output</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--USE_JDK_INFLATER">--USE_JDK_INFLATER </a>
			 / <small>-use_jdk_inflater</small>
		</h3>
		<p class="args">
			<b>Use the JDK Inflater instead of the Intel Inflater for reading compressed input</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--USE_UNPAIRED_CLIPPED_END">--USE_UNPAIRED_CLIPPED_END </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Use position of the clipping as the end position, when considering duplicates (or use the unclipped end position) (for this argument, "read end" means 3' end).</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--VALIDATION_STRINGENCY">--VALIDATION_STRINGENCY </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Validation stringency for all SAM files read by this program.  Setting stringency to SILENT can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded.</b><br />
			
		</p>
			<p>
				The --VALIDATION_STRINGENCY argument is an enumerated type (ValidationStringency), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">STRICT</dt>
					<dd class="enum"></dd>
					<dt class="enum">LENIENT</dt>
					<dd class="enum"></dd>
					<dt class="enum">SILENT</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">ValidationStringency</span>
			&nbsp;<span class="label">STRICT</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--VERBOSITY">--VERBOSITY </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Control verbosity of logging.</b><br />
			
		</p>
			<p>
				The --VERBOSITY argument is an enumerated type (LogLevel), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">ERROR</dt>
					<dd class="enum"></dd>
					<dt class="enum">WARNING</dt>
					<dd class="enum"></dd>
					<dt class="enum">INFO</dt>
					<dd class="enum"></dd>
					<dt class="enum">DEBUG</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">LogLevel</span>
			&nbsp;<span class="label">INFO</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--version">--version </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>display the version number for this tool</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>

        <hr>
        <p><a href='#top'><i class='fa fa-chevron-up'></i> Return to top</a></p>
        <hr>
        <p class="see-also">See also 
        	<a href="https://www.broadinstitute.org/picard/guide/">Picard Documentation Index</a> |
        	<a href="index.html">Tool Docs Index</a> |
        	<a href="http://gatkforums.broadinstitute.org/">Support Forum</a>
        </p>

        <p class="version">Picard version 3.0.0-17-g83b016e7f-SNAPSHOT built at 27-38-2023 01:38:51.
        </p>

	</div>

	<?php printFooter($module); ?>