<?php

    include '../../../common/include/common.php';
    include_once '../../config.php';
    $module = modules::GATK;
    printHeader($module, "Picard | Tool Documentation Index", "Guide");
?>

<link type='text/css' rel='stylesheet' href='picardDoc.css'>

<div class='row-fluid' id="top">

	

	<?php $group = 'Diagnostics and Quality Control'; ?>

	<section class="span4">
		<aside class="well">
			<a href="index.html"><h4><i class='fa fa-chevron-left'></i> Back to Tool Docs Index</h4></a>
		</aside>
		<aside class="well">
			<h2>Categories</h2>
        <style>
            #sidenav .accordion-body a {
                color : gray;
            }

            .accordion-body li {
                list-style : none;
            }
        </style>
        <ul class="nav nav-pills nav-stacked" id="sidenav">
        		<hr>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#BaseCalling">Base Calling</a>
							<div id="BaseCalling"
								<?php echo ($group == 'Base Calling')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_illumina_CheckIlluminaDirectory.html">CheckIlluminaDirectory</a>
											</li>
											<li>
												<a href="picard_illumina_CollectIlluminaBasecallingMetrics.html">CollectIlluminaBasecallingMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_CollectIlluminaLaneMetrics.html">CollectIlluminaLaneMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_ExtractIlluminaBarcodes.html">ExtractIlluminaBarcodes</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaBasecallsToFastq.html">IlluminaBasecallsToFastq</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaBasecallsToSam.html">IlluminaBasecallsToSam</a>
											</li>
											<li>
												<a href="picard_illumina_MarkIlluminaAdapters.html">MarkIlluminaAdapters</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#DiagnosticsandQualityControl">Diagnostics and Quality Control</a>
							<div id="DiagnosticsandQualityControl"
								<?php echo ($group == 'Diagnostics and Quality Control')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_AccumulateQualityYieldMetrics.html">AccumulateQualityYieldMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_AccumulateVariantCallingMetrics.html">AccumulateVariantCallingMetrics</a>
											</li>
											<li>
												<a href="picard_sam_BamIndexStats.html">BamIndexStats</a>
											</li>
											<li>
												<a href="picard_fingerprint_CalculateFingerprintMetrics.html">CalculateFingerprintMetrics</a>
											</li>
											<li>
												<a href="picard_sam_CalculateReadGroupChecksum.html">CalculateReadGroupChecksum</a>
											</li>
											<li>
												<a href="picard_fingerprint_CheckFingerprint.html">CheckFingerprint</a>
											</li>
											<li>
												<a href="picard_sam_CheckTerminatorBlock.html">CheckTerminatorBlock</a>
											</li>
											<li>
												<a href="picard_fingerprint_ClusterCrosscheckMetrics.html">ClusterCrosscheckMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectAlignmentSummaryMetrics.html">CollectAlignmentSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_CollectArraysVariantCallingMetrics.html">CollectArraysVariantCallingMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectBaseDistributionByCycle.html">CollectBaseDistributionByCycle</a>
											</li>
											<li>
												<a href="picard_analysis_CollectGcBiasMetrics.html">CollectGcBiasMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_quality_CollectHiSeqXPfFailMetrics.html">CollectHiSeqXPfFailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_directed_CollectHsMetrics.html">CollectHsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_replicates_CollectIndependentReplicateMetrics.html">CollectIndependentReplicateMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectInsertSizeMetrics.html">CollectInsertSizeMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectJumpingLibraryMetrics.html">CollectJumpingLibraryMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectMultipleMetrics.html">CollectMultipleMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectOxoGMetrics.html">CollectOxoGMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectQualityYieldMetrics.html">CollectQualityYieldMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRawWgsMetrics.html">CollectRawWgsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRnaSeqMetrics.html">CollectRnaSeqMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRrbsMetrics.html">CollectRrbsMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_CollectSamErrorMetrics.html">CollectSamErrorMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_CollectSequencingArtifactMetrics.html">CollectSequencingArtifactMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_directed_CollectTargetedPcrMetrics.html">CollectTargetedPcrMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_CollectVariantCallingMetrics.html">CollectVariantCallingMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectWgsMetrics.html">CollectWgsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectWgsMetricsWithNonZeroCoverage.html">CollectWgsMetricsWithNonZeroCoverage</a>
											</li>
											<li>
												<a href="picard_analysis_CompareMetrics.html">CompareMetrics</a>
											</li>
											<li>
												<a href="picard_sam_CompareSAMs.html">CompareSAMs</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_ConvertSequencingArtifactToOxoG.html">ConvertSequencingArtifactToOxoG</a>
											</li>
											<li>
												<a href="picard_fingerprint_CrosscheckFingerprints.html">CrosscheckFingerprints</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_EstimateLibraryComplexity.html">EstimateLibraryComplexity</a>
											</li>
											<li>
												<a href="picard_analysis_MeanQualityByCycle.html">MeanQualityByCycle</a>
											</li>
											<li>
												<a href="picard_analysis_QualityScoreDistribution.html">QualityScoreDistribution</a>
											</li>
											<li>
												<a href="picard_sam_ValidateSamFile.html">ValidateSamFile</a>
											</li>
											<li>
												<a href="picard_sam_ViewSam.html">ViewSam</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#GenotypingArraysManipulation">Genotyping Arrays Manipulation</a>
							<div id="GenotypingArraysManipulation"
								<?php echo ($group == 'Genotyping Arrays Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_arrays_illumina_BpmToNormalizationManifestCsv.html">BpmToNormalizationManifestCsv</a>
											</li>
											<li>
												<a href="picard_arrays_CombineGenotypingArrayVcfs.html">CombineGenotypingArrayVcfs</a>
											</li>
											<li>
												<a href="picard_arrays_CreateBafRegressMetricsFile.html">CreateBafRegressMetricsFile</a>
											</li>
											<li>
												<a href="picard_arrays_CreateVerifyIDIntensityContaminationMetricsFile.html">CreateVerifyIDIntensityContaminationMetricsFile</a>
											</li>
											<li>
												<a href="picard_arrays_GtcToVcf.html">GtcToVcf</a>
											</li>
											<li>
												<a href="picard_arrays_MergePedIntoVcf.html">MergePedIntoVcf</a>
											</li>
											<li>
												<a href="picard_arrays_VcfToAdpc.html">VcfToAdpc</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#IntervalsManipulation">Intervals Manipulation</a>
							<div id="IntervalsManipulation"
								<?php echo ($group == 'Intervals Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_BedToIntervalList.html">BedToIntervalList</a>
											</li>
											<li>
												<a href="picard_util_IntervalListToBed.html">IntervalListToBed</a>
											</li>
											<li>
												<a href="picard_util_IntervalListTools.html">IntervalListTools</a>
											</li>
											<li>
												<a href="picard_util_LiftOverIntervalList.html">LiftOverIntervalList</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Other">Other</a>
							<div id="Other"
								<?php echo ($group == 'Other')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_FifoBuffer.html">FifoBuffer</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReadDataManipulation">Read Data Manipulation</a>
							<div id="ReadDataManipulation"
								<?php echo ($group == 'Read Data Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_sam_AddCommentsToBam.html">AddCommentsToBam</a>
											</li>
											<li>
												<a href="picard_sam_AddOATag.html">AddOATag</a>
											</li>
											<li>
												<a href="picard_sam_AddOrReplaceReadGroups.html">AddOrReplaceReadGroups</a>
											</li>
											<li>
												<a href="picard_fastq_BamToBfq.html">BamToBfq</a>
											</li>
											<li>
												<a href="picard_sam_BuildBamIndex.html">BuildBamIndex</a>
											</li>
											<li>
												<a href="picard_sam_CleanSam.html">CleanSam</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_CollectDuplicateMetrics.html">CollectDuplicateMetrics</a>
											</li>
											<li>
												<a href="picard_sam_DownsampleSam.html">DownsampleSam</a>
											</li>
											<li>
												<a href="picard_sam_FastqToSam.html">FastqToSam</a>
											</li>
											<li>
												<a href="picard_sam_FilterSamReads.html">FilterSamReads</a>
											</li>
											<li>
												<a href="picard_sam_FixMateInformation.html">FixMateInformation</a>
											</li>
											<li>
												<a href="picard_sam_GatherBamFiles.html">GatherBamFiles</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_MarkDuplicates.html">MarkDuplicates</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_MarkDuplicatesWithMateCigar.html">MarkDuplicatesWithMateCigar</a>
											</li>
											<li>
												<a href="picard_sam_MergeBamAlignment.html">MergeBamAlignment</a>
											</li>
											<li>
												<a href="picard_sam_MergeSamFiles.html">MergeSamFiles</a>
											</li>
											<li>
												<a href="picard_sam_PositionBasedDownsampleSam.html">PositionBasedDownsampleSam</a>
											</li>
											<li>
												<a href="picard_sam_ReorderSam.html">ReorderSam</a>
											</li>
											<li>
												<a href="picard_sam_ReplaceSamHeader.html">ReplaceSamHeader</a>
											</li>
											<li>
												<a href="picard_sam_RevertOriginalBaseQualitiesAndAddMateCigar.html">RevertOriginalBaseQualitiesAndAddMateCigar</a>
											</li>
											<li>
												<a href="picard_sam_RevertSam.html">RevertSam</a>
											</li>
											<li>
												<a href="picard_sam_SamFormatConverter.html">SamFormatConverter</a>
											</li>
											<li>
												<a href="picard_sam_SamToFastq.html">SamToFastq</a>
											</li>
											<li>
												<a href="picard_sam_SetNmAndUqTags.html">SetNmAndUqTags</a>
											</li>
											<li>
												<a href="picard_sam_SetNmMdAndUqTags.html">SetNmMdAndUqTags</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_SimpleMarkDuplicatesWithMateCigar.html">SimpleMarkDuplicatesWithMateCigar</a>
											</li>
											<li>
												<a href="picard_sam_SortSam.html">SortSam</a>
											</li>
											<li>
												<a href="picard_sam_SplitSamByLibrary.html">SplitSamByLibrary</a>
											</li>
											<li>
												<a href="picard_sam_SplitSamByNumberOfReads.html">SplitSamByNumberOfReads</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_UmiAwareMarkDuplicatesWithMateCigar.html">UmiAwareMarkDuplicatesWithMateCigar</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Reference">Reference</a>
							<div id="Reference"
								<?php echo ($group == 'Reference')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_BaitDesigner.html">BaitDesigner</a>
											</li>
											<li>
												<a href="picard_sam_CreateSequenceDictionary.html">CreateSequenceDictionary</a>
											</li>
											<li>
												<a href="picard_reference_ExtractSequences.html">ExtractSequences</a>
											</li>
											<li>
												<a href="picard_reference_NonNFastaSize.html">NonNFastaSize</a>
											</li>
											<li>
												<a href="picard_reference_NormalizeFasta.html">NormalizeFasta</a>
											</li>
											<li>
												<a href="picard_util_ScatterIntervalsByNs.html">ScatterIntervalsByNs</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantEvaluationandRefinement">Variant Evaluation and Refinement</a>
							<div id="VariantEvaluationandRefinement"
								<?php echo ($group == 'Variant Evaluation and Refinement')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_vcf_MendelianViolations_FindMendelianViolations.html">FindMendelianViolations</a>
											</li>
											<li>
												<a href="picard_vcf_GenotypeConcordance.html">GenotypeConcordance</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantFiltering">Variant Filtering</a>
							<div id="VariantFiltering"
								<?php echo ($group == 'Variant Filtering')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_vcf_filter_FilterVcf.html">FilterVcf</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantManipulation">Variant Manipulation</a>
							<div id="VariantManipulation"
								<?php echo ($group == 'Variant Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_vcf_FixVcfHeader.html">FixVcfHeader</a>
											</li>
											<li>
												<a href="picard_vcf_GatherVcfs.html">GatherVcfs</a>
											</li>
											<li>
												<a href="picard_vcf_LiftoverVcf.html">LiftoverVcf</a>
											</li>
											<li>
												<a href="picard_vcf_MakeSitesOnlyVcf.html">MakeSitesOnlyVcf</a>
											</li>
											<li>
												<a href="picard_vcf_MergeVcfs.html">MergeVcfs</a>
											</li>
											<li>
												<a href="picard_vcf_RenameSampleInVcf.html">RenameSampleInVcf</a>
											</li>
											<li>
												<a href="picard_vcf_SortVcf.html">SortVcf</a>
											</li>
											<li>
												<a href="picard_vcf_SplitVcfs.html">SplitVcfs</a>
											</li>
											<li>
												<a href="picard_vcf_UpdateVcfSequenceDictionary.html">UpdateVcfSequenceDictionary</a>
											</li>
											<li>
												<a href="picard_vcf_VcfFormatConverter.html">VcfFormatConverter</a>
											</li>
											<li>
												<a href="picard_vcf_VcfToIntervalList.html">VcfToIntervalList</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Metrics">Metrics</a>
							<div id="Metrics"
								<?php echo ($group == 'Metrics')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_analysis_AlignmentSummaryMetrics.html">AlignmentSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_CollectArraysVariantCallingMetrics$ArraysControlCodesSummaryMetrics.html">ArraysControlCodesSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_CollectArraysVariantCallingMetrics$ArraysVariantCallingDetailMetrics.html">ArraysVariantCallingDetailMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_CollectArraysVariantCallingMetrics$ArraysVariantCallingSummaryMetrics.html">ArraysVariantCallingSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_BafRegressMetrics.html">BafRegressMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_SequencingArtifactMetrics$BaitBiasDetailMetrics.html">BaitBiasDetailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_SequencingArtifactMetrics$BaitBiasSummaryMetrics.html">BaitBiasSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_BarcodeMetric.html">BarcodeMetric</a>
											</li>
											<li>
												<a href="picard_analysis_BaseDistributionByCycleMetrics.html">BaseDistributionByCycleMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_BaseErrorMetric.html">BaseErrorMetric</a>
											</li>
											<li>
												<a href="picard_fingerprint_ClusteredCrosscheckMetric.html">ClusteredCrosscheckMetric</a>
											</li>
											<li>
												<a href="picard_analysis_CollectOxoGMetrics$CpcgMetrics.html">CpcgMetrics</a>
											</li>
											<li>
												<a href="picard_sam_DuplicationMetrics.html">DuplicationMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_ErrorMetric.html">ErrorMetric</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_ErrorSummaryMetrics.html">ErrorSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_fingerprint_FingerprintMetrics.html">FingerprintMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_FingerprintingDetailMetrics.html">FingerprintingDetailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_FingerprintingSummaryMetrics.html">FingerprintingSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_GcBiasDetailMetrics.html">GcBiasDetailMetrics</a>
											</li>
											<li>
												<a href="picard_metrics_GcBiasMetrics.html">GcBiasMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_GcBiasSummaryMetrics.html">GcBiasSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_GenotypeConcordanceContingencyMetrics.html">GenotypeConcordanceContingencyMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_GenotypeConcordanceDetailMetrics.html">GenotypeConcordanceDetailMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_GenotypeConcordanceSummaryMetrics.html">GenotypeConcordanceSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_directed_HsMetrics.html">HsMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaBasecallingMetrics.html">IlluminaBasecallingMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaLaneMetrics.html">IlluminaLaneMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaPhasingMetrics.html">IlluminaPhasingMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_IndelErrorMetric.html">IndelErrorMetric</a>
											</li>
											<li>
												<a href="picard_analysis_replicates_IndependentReplicateMetric.html">IndependentReplicateMetric</a>
											</li>
											<li>
												<a href="picard_analysis_InsertSizeMetrics.html">InsertSizeMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_JumpingLibraryMetrics.html">JumpingLibraryMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_MendelianViolations_MendelianViolationMetrics.html">MendelianViolationMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_OverlappingErrorMetric.html">OverlappingErrorMetric</a>
											</li>
											<li>
												<a href="picard_illumina_quality_CollectHiSeqXPfFailMetrics$PFFailDetailedMetric.html">PFFailDetailedMetric</a>
											</li>
											<li>
												<a href="picard_illumina_quality_CollectHiSeqXPfFailMetrics$PFFailSummaryMetric.html">PFFailSummaryMetric</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_SequencingArtifactMetrics$PreAdapterDetailMetrics.html">PreAdapterDetailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_SequencingArtifactMetrics$PreAdapterSummaryMetrics.html">PreAdapterSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectQualityYieldMetrics$QualityYieldMetrics.html">QualityYieldMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRawWgsMetrics$RawWgsMetrics.html">RawWgsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_RnaSeqMetrics.html">RnaSeqMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_RrbsCpgDetailMetrics.html">RrbsCpgDetailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_RrbsSummaryMetrics.html">RrbsSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamComparisonMetric.html">SamComparisonMetric</a>
											</li>
											<li>
												<a href="picard_analysis_directed_TargetMetrics.html">TargetMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_directed_TargetedPcrMetrics.html">TargetedPcrMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_TheoreticalSensitivityMetrics.html">TheoreticalSensitivityMetrics</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_UmiMetrics.html">UmiMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_CollectVariantCallingMetrics$VariantCallingDetailMetrics.html">VariantCallingDetailMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_CollectVariantCallingMetrics$VariantCallingSummaryMetrics.html">VariantCallingSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_VerifyIDIntensityContaminationMetrics.html">VerifyIDIntensityContaminationMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_WgsMetrics.html">WgsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectWgsMetricsWithNonZeroCoverage$WgsMetricsWithNonZeroCoverage.html">WgsMetricsWithNonZeroCoverage</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
        </ul>
		</aside>
		<?php getForumPosts( 'CrosscheckFingerprints' ) ?>

	</section>

	<div class="span8">

		<h1>CrosscheckFingerprints</h1>

		<p class="lead">Checks the odds that all data in the set of input files come from the same individual. Can be used to cross-check readgroups, libraries, samples, or files. Acceptable inputs include BAM/SAM/CRAM and VCF/GVCF files. Output delivers LOD scores in the form of a CrosscheckMetric file. 

<h3>Summary</h3>
CrosscheckFingerprints rapidly checks the odds that all of the genetic data within a set of files come from the same individual. This is accomplished by selectively sampling from the input files, and determining whether the genotypes of the specified Groups match to each other. (Groups are defined by the input and the argument CROSSCHECK_BY; they can be READ_GROUP, LIBRARY, SAMPLE, or FILE.)<br /><br /> Output is generated in the form of a “molten” (one row per comparison) CrosscheckMetric file that includes the Logarithm of the Odds (LOD) score, as well as the tumor-aware LOD score. Tumor-aware LOD scores can be used to assess genotypic identity in the presence of a severe Loss of Heterozygosity (LOH) with high purity—this could otherwise lead to a failure of the tool to identify samples are from the same individual. Output is also available as a matrix, to facilitate visual inspection of crosscheck results.<br /><br /> Metric files can contain many rows of output. We therefore recommend following up CrosscheckFingerprints with a step using [ClusterCrosscheckMetrics (Picard)](https://gatk.broadinstitute.org/hc/en-us/articles/360045798972--Tool-Documentation-Index); this tool will cluster groups together that pass a designated LOD threshold, ensuring that groups within the cluster are related to each other. <br /><br /> There may be cases where several groups out of a collection of possible groups must be identified---for example, to link a BAM to its correct sample in a multi-sample VCF. In this case, it would not be necessary to cross-check the various samples in the VCF against each other, but only to check the identity of the BAM against the various samples in the VCF. For this application, the SECOND_INPUT argument is provided. With SECOND_INPUT, CrosscheckFingerprints can do the following: <br /><br /> <ul><li> Independently aggregate data for the input files in INPUT and SECOND_INPUT. </li><li> Aggregate data at the SAMPLE level. </li><li> Compare samples from INPUT to the same sample in SECOND_INPUT. </li><li> Disables MATRIX_OUTPUT. </li></ul><br /><br />In some cases, the groups collected may not have any observations (‘reads’ for BAM files, or ‘calls’ for VCF files) at fingerprinting sites. Alternatively, a sample in INPUT may be missing from SECOND_INPUT. These cases are handled as follows: <br /><br /> <ul><li> If running in CHECK_SAME_SAMPLES mode with the INPUT and SECOND_INPUT sets of input files: when either set of inputs (1) includes a sample not found in the other, or (2) contains a sample with no observations at any fingerprinting sites, then an error will be logged and the tool will return EXIT_CODE_WHEN_MISMATCH. </li><li> If running in any other running mode: when a group which is being crosschecked does not have any observations at fingerprinting sites, a warning will be logged. </li></ul><br /><br />Note that, as long as there is at least one comparison in which both files have observations at fingerprinting sites, the tool will return a ‘zero’. However, an error will be logged and the tool will return EXIT_CODE_WHEN_NO_VALID_CHECKS if all comparisons have at least one side without observations at a fingerprinting site (ie. all LOD scores are zero). <br /><br /> <hr/><h3>Examples</h3><h4>Check that all the readgroups from a sample match each other:</h4><pre>    java -jar picard.jar CrosscheckFingerprints \
          INPUT=sample.with.many.readgroups.bam \
          HAPLOTYPE_MAP=fingerprinting_haplotype_database.txt \
          LOD_THRESHOLD=-5 \
          OUTPUT=sample.crosscheck_metrics </pre>
 <h4>Check that all the readgroups match as expected when providing reads from two samples from the same individual:</h4> <pre>     java -jar picard.jar CrosscheckFingerprints \
           INPUT=sample.one.with.many.readgroups.bam \
           INPUT=sample.two.with.many.readgroups.bam \
           HAPLOTYPE_MAP=fingerprinting_haplotype_database.txt \
           LOD_THRESHOLD=-5 \
           EXPECT_ALL_GROUPS_TO_MATCH=true \
           OUTPUT=sample.crosscheck_metrics </pre><br /><br /><h4>Detailed Explanation</h4>
This tool calculates the LOD score for identity check between "groups" of data in the INPUT files as defined by the CROSSCHECK_BY argument. A positive value indicates that the data seems to have come from the same individual or, in other words the identity checks out. The scale is logarithmic (base 10), so a LOD of 6 indicates that it is 1,000,000 more likely that the data matches the genotypes than not. A negative value indicates that the data do not match. A score that is near zero is inconclusive and can result from low coverage or non-informative genotypes. <br /><br />Each group is assigned a sample identifier (for SAM this is taken from the SM tag in the appropriate readgroup header line, for VCF this is taken from the column label in the file-header. After combining all the data from the same group together, an all-against-all comparison is performed. Results are categorized as one of EXPECTED_MATCH, EXPECTED_MISMATCH, UNEXPECTED_MATCH, UNEXPECTED_MISMATCH, or AMBIGUOUS depending on the LOD score and on whether the sample identifiers of the groups agree: LOD scores that are less than LOD_THRESHOLD are considered mismatches, and those greater than -LOD_THRESHOLD are matches (between is ambiguous). If the sample identifiers are equal, the groups are expected to match. They are expected to mismatch otherwise. <br /><br />The identity check makes use of haplotype blocks defined in the HAPLOTYPE_MAP file to enable it to have higher statistical power for detecting identity or swap by aggregating data from several SNPs in the haplotype block. This enables an identity check of samples with very low coverage (e.g. ~1x mean coverage).<br /><br />When provided a VCF, the identity check looks at the PL, GL and GT fields (in that order) and uses the first one that it finds. </p>

			<h3>Category
				<small> Diagnostics and Quality Control</small>
			</h3>
		<hr>
		<h2>Overview</h2>
		Checks that all data in the set of input files appear to come from the same 
 individual. Can be used to compare according to readgroups, libraries, samples, or files. 
 Operates on bams/sams and vcfs (including gvcfs). 

 <h3>Summary</h3>
 Checks if all the genetic data within a set of files appear to come from the same individual. 
 It quickly determines whether a "group's" genotype matches that of an input SAM/BAM/VCF by selective sampling, 
 and has been designed to work well even for low-depth SAM/BAMs. 
 <br/>
 The tool collects "fingerprints" (essentially genotype information from different parts of the genome) 
 at the finest level available in the data (readgroup for SAM files 
 and sample for VCF files) and then optionally aggregates it by library, sample or file, to increase power and provide 
 results at the desired resolution. Output is in a "Moltenized" format, one row per comparison. The results will 
 be emitted into a metric file for the class {@link CrosscheckMetric}. 
 In this format, the output will include both the LOD score and the tumor-aware LOD score; the latter can 
 help assess identity even in the presence of a severe loss of heterozygosity with high purity (the tool could 
 otherwise fail to notice that samples are from the same individual.) 
 A matrix output is also available to facilitate visual inspection of crosscheck results. 
 <br/>
 Since there can be many rows of output in the metric file, we recommend the use of {@link ClusterCrosscheckMetrics} 
 as a follow-up step to running CrosscheckFingerprints. 
 <br/>
 There are cases where one would like to identify a few groups out of a collection of many possible groups (say 
 to link a bam to its correct sample in a multi-sample vcf. In this case, one would not care for the cross-checking 
 of the various samples in the VCF against each other, but only in checking the identity of the bam against the various 
 samples in the vcf. The {@link #SECOND_INPUT} is provided for this use-case. With {@link #SECOND_INPUT} provided, 
 CrosscheckFingerprints does the following: 
 <il>
 <li>aggregation of data happens independently for the input files in {@link #INPUT} and {@link #SECOND_INPUT}.</li>
 <li>aggregation of data happens at the SAMPLE level.</li>
 <li>each sample from {@link #INPUT} will only be compared to that same sample in {@link #INPUT}.</li>
 <li>{@link #MATRIX_OUTPUT} is disabled.</li>
 </il>
 <br/>
 In some cases, the groups collected may not have any observations (calls for a vcf, reads for a bam) at fingerprinting 
 sites, or a sample in INPUT may be missing from the SECOND_INPUT. These cases are handled as follows:  If running in 
 CHECK_SAME_SAMPLES mode with INPUT and SECOND_INPUT, and either INPUT or SECOND_INPUT includes a sample not found in the 
 other, or contains a sample with no observations at any fingerprinting sites, an error will be logged and the tool will 
 return EXIT_CODE_WHEN_MISMATCH.
 In all other running modes, when any group which is being crosschecked does not have any observations at fingerprinting 
 sites, a warning is logged.  As long as there is at least one comparison where both sides have observations at 
 fingerprinting sites, the tool will return zero.  However, if all comparisons have at least one side with no observations
 at fingerprinting sites, an error will be logged and the tool will return EXIT_CODE_WHEN_NO_VALID_CHECKS.
 <h3>Examples</h3>
 <h4>Check that all the readgroups from a sample match each other:</h4>
 <pre>
     java -jar picard.jar CrosscheckFingerprints \
          INPUT=sample.with.many.readgroups.bam \
          HAPLOTYPE_MAP=fingerprinting_haplotype_database.txt \
          LOD_THRESHOLD=-5 \
          OUTPUT=sample.crosscheck_metrics
 </pre>


 <h4>Check that all the readgroups match as expected when providing reads from two samples from the same individual:</h4>
 <pre>
     java -jar picard.jar CrosscheckFingerprints \
          INPUT=sample.one.with.many.readgroups.bam \
          INPUT=sample.two.with.many.readgroups.bam \
          HAPLOTYPE_MAP=fingerprinting_haplotype_database.txt \
          LOD_THRESHOLD=-5 \
          EXPECT_ALL_GROUPS_TO_MATCH=true \
          OUTPUT=sample.crosscheck_metrics
 </pre>


 <h3> Detailed Explanation</h3>

 This tool calculates the LOD score for identity check between "groups" of data in the INPUT files as defined by
 the CROSSCHECK_BY argument. A positive value indicates that the data seems to have come from the same individual
 or, in other words, the identity checks out. The scale is logarithmic (base 10), so a LOD of 6 indicates
 that it is 1,000,000 times more likely that the data matches the genotypes than not. A negative value indicates
 that the data do not match. A score that is near zero is inconclusive and can result from low coverage
 or non-informative genotypes. Each group is assigned a sample identifier (for SAM this is taken from the SM tag in
 the appropriate readgroup header line, for VCF this is taken from the column label in the file-header.
 After combining all the data from the same "group" together, an all-against-all comparison is performed. Results are
 categorized as a {@link FingerprintResult} enum: EXPECTED_MATCH, EXPECTED_MISMATCH, UNEXPECTED_MATCH, UNEXPECTED_MISMATCH,
 or AMBIGUOUS depending on the LOD score and on whether the sample identifiers of the groups agree: LOD scores that are
 less than LOD_THRESHOLD are considered mismatches, and those greater than -LOD_THRESHOLD are matches (between is ambiguous).
 If the sample identifiers are equal, the groups are expected to match. They are expected to mismatch otherwise.
 <br/>

 The identity check makes use of haplotype blocks defined in the HAPLOTYPE_MAP file to enable it to have higher
 statistical power for detecting identity or swap by aggregating data from several SNPs in the haplotype block. This
 enables an identity check of samples with very low coverage (e.g. ~1x mean coverage).
 <br/>
 When provided a VCF, the identity check looks at the PL, GL and GT fields (in that order) and uses the first one that
 it finds.
@author Tim Fennell
@author Yossi Farjoun


				<h3>CrosscheckFingerprints specific arguments</h3>
				<p>This table summarizes the command-line arguments that are specific to this tool. For more details on each argument, see the list further down below the table or click on an argument name to jump directly to that entry in the list.</p>
				<table class="table table-striped table-bordered table-condensed">
					<thead>
					<tr>
						<th>Argument name(s)</th>
						<th>Default value</th>
						<th>Summary</th>
					</tr>
					</thead>
					<tbody>
			<tr>
				<th colspan="4" id="row-divider">Required Arguments</th>
			</tr>
				<tr>
					<td><a href="#--HAPLOTYPE_MAP">--HAPLOTYPE_MAP</a><br />
								&nbsp;<em>-H</em>
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>The file lists a set of SNPs, optionally arranged in high-LD blocks, to be used for fingerprinting. See https://software.broadinstitute.org/gatk/documentation/article?id=9526 for details.</td>
				</tr>
				<tr>
					<td><a href="#--INPUT">--INPUT</a><br />
								&nbsp;<em>-I</em>
					</td>
					<!--<td>List[String]</td> -->
					<td>[]</td>
					<td>One or more input files (or lists of files) with which to compare fingerprints.</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Tool Arguments</th>
			</tr>
				<tr>
					<td><a href="#--ALLOW_DUPLICATE_READS">--ALLOW_DUPLICATE_READS</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Allow the use of duplicate reads in performing the comparison. Can be useful when duplicate marking has been overly aggressive and coverage is low.</td>
				</tr>
				<tr>
					<td><a href="#--arguments_file">--arguments_file</a><br />
					</td>
					<!--<td>List[File]</td> -->
					<td>[]</td>
					<td>read one or more arguments files and add them to the command line</td>
				</tr>
				<tr>
					<td><a href="#--CALCULATE_TUMOR_AWARE_RESULTS">--CALCULATE_TUMOR_AWARE_RESULTS</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>true</td>
					<td>Specifies whether the Tumor-aware result should be calculated. These are time consuming and can roughly double the runtime of the tool. When crosschecking many groups not calculating the tumor-aware  results can result in a significant speedup.</td>
				</tr>
				<tr>
					<td><a href="#--CROSSCHECK_BY">--CROSSCHECK_BY</a><br />
					</td>
					<!--<td>DataType</td> -->
					<td>READGROUP</td>
					<td>Specifies which data-type should be used as the basic comparison unit. Fingerprints from readgroups can be "rolled-up" to the LIBRARY, SAMPLE, or FILE level before being compared. Fingerprints from VCF can be be compared by SAMPLE or FILE.</td>
				</tr>
				<tr>
					<td><a href="#--CROSSCHECK_MODE">--CROSSCHECK_MODE</a><br />
					</td>
					<!--<td>CrosscheckMode</td> -->
					<td>CHECK_SAME_SAMPLE</td>
					<td>An argument that controls how crosschecking with both INPUT and SECOND_INPUT should occur. </td>
				</tr>
				<tr>
					<td><a href="#--EXIT_CODE_WHEN_MISMATCH">--EXIT_CODE_WHEN_MISMATCH</a><br />
					</td>
					<!--<td>int</td> -->
					<td>1</td>
					<td>When one or more mismatches between groups is detected, exit with this value instead of 0.</td>
				</tr>
				<tr>
					<td><a href="#--EXIT_CODE_WHEN_NO_VALID_CHECKS">--EXIT_CODE_WHEN_NO_VALID_CHECKS</a><br />
					</td>
					<!--<td>int</td> -->
					<td>1</td>
					<td>When all LOD scores are zero, exit with this value.</td>
				</tr>
				<tr>
					<td><a href="#--EXPECT_ALL_GROUPS_TO_MATCH">--EXPECT_ALL_GROUPS_TO_MATCH</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Expect all groups' fingerprints to match, irrespective of their sample names.  By default (with this value set to false), groups (readgroups, libraries, files, or samples) with different sample names are expected to mismatch, and those with the same sample name are expected to match. </td>
				</tr>
				<tr>
					<td><a href="#--GENOTYPING_ERROR_RATE">--GENOTYPING_ERROR_RATE</a><br />
					</td>
					<!--<td>double</td> -->
					<td>0.01</td>
					<td>Assumed genotyping error rate that provides a floor on the probability that a genotype comes from the expected sample. Must be greater than zero. </td>
				</tr>
				<tr>
					<td><a href="#--help">--help</a><br />
								&nbsp;<em>-h</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>display the help message</td>
				</tr>
				<tr>
					<td><a href="#--INPUT_INDEX_MAP">--INPUT_INDEX_MAP</a><br />
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>A tsv with two columns and no header which maps the input files to corresponding indices; to be used when index files are not located next to input files. First column must match the list of inputs. </td>
				</tr>
				<tr>
					<td><a href="#--INPUT_SAMPLE_FILE_MAP">--INPUT_SAMPLE_FILE_MAP</a><br />
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>A tsv with two columns representing the sample as it should be used for comparisons to SECOND_INPUT (in the first column) and  the source file (in INPUT) for the fingerprint (in the second column). Need only to include the samples that change. Values in column 1 should be unique even in union with the remaining unmapped samples. Values in column 2 should be unique in the file. Will error if more than one sample is found in a file (multi-sample VCF) pointed to in column 2. Should only be used in the presence of SECOND_INPUT. </td>
				</tr>
				<tr>
					<td><a href="#--INPUT_SAMPLE_MAP">--INPUT_SAMPLE_MAP</a><br />
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>A tsv with two columns representing the sample as it appears in the INPUT data (in column 1) and the sample as it should be used for comparisons to SECOND_INPUT (in the second column). Need only include the samples that change. Values in column 1 should be unique. Values in column 2 should be unique even in union with the remaining unmapped samples. Should only be used with SECOND_INPUT. </td>
				</tr>
				<tr>
					<td><a href="#--LOD_THRESHOLD">--LOD_THRESHOLD</a><br />
								&nbsp;<em>-LOD</em>
					</td>
					<!--<td>double</td> -->
					<td>0.0</td>
					<td>If any two groups (with the same sample name) match with a LOD score lower than the threshold the tool will exit with a non-zero code to indicate error. Program will also exit with an error if it finds two groups with different sample name that match with a LOD score greater than -LOD_THRESHOLD.

LOD score 0 means equal likelihood that the groups match vs. come from different individuals, negative LOD score -N, mean 10^N time more likely that the groups are from different individuals, and +N means 10^N times more likely that the groups are from the same individual. </td>
				</tr>
				<tr>
					<td><a href="#--LOSS_OF_HET_RATE">--LOSS_OF_HET_RATE</a><br />
					</td>
					<!--<td>double</td> -->
					<td>0.5</td>
					<td>The rate at which a heterozygous genotype in a normal sample turns into a homozygous (via loss of heterozygosity) in the tumor (model assumes independent events, so this needs to be larger than reality).</td>
				</tr>
				<tr>
					<td><a href="#--MATRIX_OUTPUT">--MATRIX_OUTPUT</a><br />
								&nbsp;<em>-MO</em>
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>Optional output file to write matrix of LOD scores to. This is less informative than the metrics output and only contains Normal-Normal LOD score (i.e. doesn't account for Loss of Heterozygosity). It is however sometimes easier to use visually.</td>
				</tr>
				<tr>
					<td><a href="#--MAX_EFFECT_OF_EACH_HAPLOTYPE_BLOCK">--MAX_EFFECT_OF_EACH_HAPLOTYPE_BLOCK</a><br />
					</td>
					<!--<td>double</td> -->
					<td>3.0</td>
					<td>Maximal effect of any single haplotype block on outcome (-log10 of maximal likelihood difference between the different values for the three possible genotypes).</td>
				</tr>
				<tr>
					<td><a href="#--NUM_THREADS">--NUM_THREADS</a><br />
					</td>
					<!--<td>int</td> -->
					<td>1</td>
					<td>The number of threads to use to process files and generate fingerprints.</td>
				</tr>
				<tr>
					<td><a href="#--OUTPUT">--OUTPUT</a><br />
								&nbsp;<em>-O</em>
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>Optional output file to write metrics to. Default is to write to stdout.</td>
				</tr>
				<tr>
					<td><a href="#--OUTPUT_ERRORS_ONLY">--OUTPUT_ERRORS_ONLY</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>If true, then only groups that do not relate to each other as expected will have their LODs reported.</td>
				</tr>
				<tr>
					<td><a href="#--REQUIRE_INDEX_FILES">--REQUIRE_INDEX_FILES</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>A boolean value to determine whether input files should only be parsed if index files are available. Without turning this option on, the tool will need to read through the entirety of input files without index files either provided via the INPUT_INDEX_MAP or locally accessible relative to the input, which significantly increases runtime. If set to true and no index is found for a file, an exception will be thrown. This applies for both the INPUT and SECOND_INPUT files.</td>
				</tr>
				<tr>
					<td><a href="#--SAMPLE_INDIVIDUAL_MAP">--SAMPLE_INDIVIDUAL_MAP</a><br />
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>A tsv with two columns representing the individual with which each sample is associated.  The first column is the sample id, and the second column is the associated individual id.  Values in the first column must be unique. If INPUT_SAMPLE_MAP or SECOND_INPUT_SAMPLE_MAP is also specified, then the values in the first column of this file should be the sample aliases specified in the second columns of INPUT_SAMPLE_MAP and SECOND_INPUT_SAMPLE_MAP, respectively.  When this input is specified, expectations for matches will be based on the equality or inequality of the individual ids associated with two samples, as opposed to the sample ids themselves.  Samples which are not listed in this file will have their sample id used as their individual id, for the purposes of match expectations.  This means that one sample id could be used as the individual id for another sample, but not included in the map itself, and these two samples would be considered to have come from the same individual.  Note that use of this parameter only affects labelling of matches and mismatches as EXPECTED or UNEXPECTED.  It has no affect on how data is grouped for crosschecking.</td>
				</tr>
				<tr>
					<td><a href="#--SECOND_INPUT">--SECOND_INPUT</a><br />
								&nbsp;<em>-SI</em>
					</td>
					<!--<td>List[String]</td> -->
					<td>[]</td>
					<td>A second set of input files (or lists of files) with which to compare fingerprints. If this option is provided the tool compares each sample in INPUT with the sample from SECOND_INPUT that has the same sample ID. In addition, data will be grouped by SAMPLE regardless of the value of CROSSCHECK_BY. When operating in this mode, each sample in INPUT must also have a corresponding sample in SECOND_INPUT. If this is violated, the tool will proceed to check the matching samples, but report the missing samples and return a non-zero error-code.</td>
				</tr>
				<tr>
					<td><a href="#--SECOND_INPUT_INDEX_MAP">--SECOND_INPUT_INDEX_MAP</a><br />
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>A tsv with two columns and no header which maps the second input files to corresponding indices; to be used when index files are not located next to second input files. First column must match the list of second inputs. </td>
				</tr>
				<tr>
					<td><a href="#--SECOND_INPUT_SAMPLE_MAP">--SECOND_INPUT_SAMPLE_MAP</a><br />
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>A tsv with two columns representing the sample as it appears in the SECOND_INPUT data (in column 1) and the sample as it should be used for comparisons to INPUT (in the second column). Note that in case of unrolling files (file-of-filenames) one would need to reference the final file, i.e. the file that contains the genomic data. Need only include the samples that change. Values in column 1 should be unique. Values in column 2 should be unique even in union with the remaining unmapped samples. Should only be used with SECOND_INPUT. </td>
				</tr>
				<tr>
					<td><a href="#--version">--version</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>display the version number for this tool</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Common Arguments</th>
			</tr>
				<tr>
					<td><a href="#--COMPRESSION_LEVEL">--COMPRESSION_LEVEL</a><br />
					</td>
					<!--<td>int</td> -->
					<td>5</td>
					<td>Compression level for all compressed files created (e.g. BAM and VCF).</td>
				</tr>
				<tr>
					<td><a href="#--CREATE_INDEX">--CREATE_INDEX</a><br />
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Whether to create an index when writing VCF or coordinate sorted BAM output.</td>
				</tr>
				<tr>
					<td><a href="#--CREATE_MD5_FILE">--CREATE_MD5_FILE</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Whether to create an MD5 digest for any BAM or FASTQ files created.  </td>
				</tr>
				<tr>
					<td><a href="#--MAX_RECORDS_IN_RAM">--MAX_RECORDS_IN_RAM</a><br />
					</td>
					<!--<td>Integer</td> -->
					<td>500000</td>
					<td>When writing files that need to be sorted, this will specify the number of records stored in RAM before spilling to disk. Increasing this number reduces the number of file handles needed to sort the file, and increases the amount of RAM needed.</td>
				</tr>
				<tr>
					<td><a href="#--QUIET">--QUIET</a><br />
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Whether to suppress job-summary info on System.err.</td>
				</tr>
				<tr>
					<td><a href="#--REFERENCE_SEQUENCE">--REFERENCE_SEQUENCE</a><br />
								&nbsp;<em>-R</em>
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>Reference sequence file.</td>
				</tr>
				<tr>
					<td><a href="#--TMP_DIR">--TMP_DIR</a><br />
					</td>
					<!--<td>List[File]</td> -->
					<td>[]</td>
					<td>One or more directories with space available to be used by this program for temporary storage of working files</td>
				</tr>
				<tr>
					<td><a href="#--USE_JDK_DEFLATER">--USE_JDK_DEFLATER</a><br />
								&nbsp;<em>-use_jdk_deflater</em>
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Use the JDK Deflater instead of the Intel Deflater for writing compressed output</td>
				</tr>
				<tr>
					<td><a href="#--USE_JDK_INFLATER">--USE_JDK_INFLATER</a><br />
								&nbsp;<em>-use_jdk_inflater</em>
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Use the JDK Inflater instead of the Intel Inflater for reading compressed input</td>
				</tr>
				<tr>
					<td><a href="#--VALIDATION_STRINGENCY">--VALIDATION_STRINGENCY</a><br />
					</td>
					<!--<td>ValidationStringency</td> -->
					<td>STRICT</td>
					<td>Validation stringency for all SAM files read by this program.  Setting stringency to SILENT can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded.</td>
				</tr>
				<tr>
					<td><a href="#--VERBOSITY">--VERBOSITY</a><br />
					</td>
					<!--<td>LogLevel</td> -->
					<td>INFO</td>
					<td>Control verbosity of logging.</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Advanced Arguments</th>
			</tr>
				<tr>
					<td><a href="#--showHidden">--showHidden</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>display hidden arguments</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Deprecated Arguments</th>
			</tr>
				<tr>
					<td><a href="#--GENOTYPING_ERROR_RATE">--GENOTYPING_ERROR_RATE</a><br />
					</td>
					<!--<td>double</td> -->
					<td>0.01</td>
					<td>Assumed genotyping error rate that provides a floor on the probability that a genotype comes from the expected sample. Must be greater than zero. </td>
				</tr>
					</tbody>
				</table>

					<h3>Argument details</h3>
					<p>Arguments in this list are specific to this tool. Keep in mind that other arguments are available that are shared with other tools (e.g. command-line GATK arguments); see Inherited arguments above.</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--ALLOW_DUPLICATE_READS">--ALLOW_DUPLICATE_READS </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Allow the use of duplicate reads in performing the comparison. Can be useful when duplicate marking has been overly aggressive and coverage is low.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--arguments_file">--arguments_file </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>read one or more arguments files and add them to the command line</b><br />
			
		</p>
		<p>
		<span class="label label-info ">List[File]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--CALCULATE_TUMOR_AWARE_RESULTS">--CALCULATE_TUMOR_AWARE_RESULTS </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Specifies whether the Tumor-aware result should be calculated. These are time consuming and can roughly double the runtime of the tool. When crosschecking many groups not calculating the tumor-aware  results can result in a significant speedup.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">true</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--COMPRESSION_LEVEL">--COMPRESSION_LEVEL </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Compression level for all compressed files created (e.g. BAM and VCF).</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">5</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--CREATE_INDEX">--CREATE_INDEX </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Whether to create an index when writing VCF or coordinate sorted BAM output.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--CREATE_MD5_FILE">--CREATE_MD5_FILE </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Whether to create an MD5 digest for any BAM or FASTQ files created.  </b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--CROSSCHECK_BY">--CROSSCHECK_BY </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Specifies which data-type should be used as the basic comparison unit. Fingerprints from readgroups can be "rolled-up" to the LIBRARY, SAMPLE, or FILE level before being compared. Fingerprints from VCF can be be compared by SAMPLE or FILE.</b><br />
			
		</p>
			<p>
				The --CROSSCHECK_BY argument is an enumerated type (DataType), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">FILE</dt>
					<dd class="enum"></dd>
					<dt class="enum">SAMPLE</dt>
					<dd class="enum"></dd>
					<dt class="enum">LIBRARY</dt>
					<dd class="enum"></dd>
					<dt class="enum">READGROUP</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">DataType</span>
			&nbsp;<span class="label">READGROUP</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--CROSSCHECK_MODE">--CROSSCHECK_MODE </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>An argument that controls how crosschecking with both INPUT and SECOND_INPUT should occur. </b><br />
			
		</p>
			<p>
				The --CROSSCHECK_MODE argument is an enumerated type (CrosscheckMode), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">CHECK_SAME_SAMPLE</dt>
					<dd class="enum">In this mode, each sample in INPUT will only be checked against a single corresponding sample in SECOND_INPUT. If a corresponding sample cannot be found, the program will proceed, but report the missing samples and return the value specified in EXIT_CODE_WHEN_MISMATCH. The corresponding samples are those that equal each other, after possible renaming via INPUT_SAMPLE_MAP and SECOND_INPUT_SAMPLE_MAP. In this mode CROSSCHECK_BY must be SAMPLE.</dd>
					<dt class="enum">CHECK_ALL_OTHERS</dt>
					<dd class="enum">In this mode, each sample in INPUT will be checked against all the samples in SECOND_INPUT.</dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">CrosscheckMode</span>
			&nbsp;<span class="label">CHECK_SAME_SAMPLE</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--EXIT_CODE_WHEN_MISMATCH">--EXIT_CODE_WHEN_MISMATCH </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>When one or more mismatches between groups is detected, exit with this value instead of 0.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">1</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--EXIT_CODE_WHEN_NO_VALID_CHECKS">--EXIT_CODE_WHEN_NO_VALID_CHECKS </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>When all LOD scores are zero, exit with this value.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">1</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--EXPECT_ALL_GROUPS_TO_MATCH">--EXPECT_ALL_GROUPS_TO_MATCH </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Expect all groups' fingerprints to match, irrespective of their sample names.  By default (with this value set to false), groups (readgroups, libraries, files, or samples) with different sample names are expected to mismatch, and those with the same sample name are expected to match. </b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--GENOTYPING_ERROR_RATE">--GENOTYPING_ERROR_RATE </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Assumed genotyping error rate that provides a floor on the probability that a genotype comes from the expected sample. Must be greater than zero. </b><br />
			
		</p>
		<p>
		<span class="label label-info ">double</span>
			&nbsp;<span class="label">0.01</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--HAPLOTYPE_MAP">--HAPLOTYPE_MAP </a>
			 / <small>-H</small>
		</h3>
		<p class="args">
			<b>The file lists a set of SNPs, optionally arranged in high-LD blocks, to be used for fingerprinting. See https://software.broadinstitute.org/gatk/documentation/article?id=9526 for details.</b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--help">--help </a>
			 / <small>-h</small>
		</h3>
		<p class="args">
			<b>display the help message</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--INPUT">--INPUT </a>
			 / <small>-I</small>
		</h3>
		<p class="args">
			<b>One or more input files (or lists of files) with which to compare fingerprints.</b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">List[String]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--INPUT_INDEX_MAP">--INPUT_INDEX_MAP </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>A tsv with two columns and no header which maps the input files to corresponding indices; to be used when index files are not located next to input files. First column must match the list of inputs. </b><br />
			
		</p>
		<p>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--INPUT_SAMPLE_FILE_MAP">--INPUT_SAMPLE_FILE_MAP </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>A tsv with two columns representing the sample as it should be used for comparisons to SECOND_INPUT (in the first column) and  the source file (in INPUT) for the fingerprint (in the second column). Need only to include the samples that change. Values in column 1 should be unique even in union with the remaining unmapped samples. Values in column 2 should be unique in the file. Will error if more than one sample is found in a file (multi-sample VCF) pointed to in column 2. Should only be used in the presence of SECOND_INPUT. </b><br />
			
		</p>
			<p><b>Exclusion:</b> This argument cannot be used at the same time as <code>INPUT_SAMPLE_MAP</code>.</p>
		<p>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--INPUT_SAMPLE_MAP">--INPUT_SAMPLE_MAP </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>A tsv with two columns representing the sample as it appears in the INPUT data (in column 1) and the sample as it should be used for comparisons to SECOND_INPUT (in the second column). Need only include the samples that change. Values in column 1 should be unique. Values in column 2 should be unique even in union with the remaining unmapped samples. Should only be used with SECOND_INPUT. </b><br />
			
		</p>
			<p><b>Exclusion:</b> This argument cannot be used at the same time as <code>INPUT_SAMPLE_FILE_MAP</code>.</p>
		<p>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--LOD_THRESHOLD">--LOD_THRESHOLD </a>
			 / <small>-LOD</small>
		</h3>
		<p class="args">
			<b>If any two groups (with the same sample name) match with a LOD score lower than the threshold the tool will exit with a non-zero code to indicate error. Program will also exit with an error if it finds two groups with different sample name that match with a LOD score greater than -LOD_THRESHOLD.

LOD score 0 means equal likelihood that the groups match vs. come from different individuals, negative LOD score -N, mean 10^N time more likely that the groups are from different individuals, and +N means 10^N times more likely that the groups are from the same individual. </b><br />
			
		</p>
		<p>
		<span class="label label-info ">double</span>
			&nbsp;<span class="label">0.0</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--LOSS_OF_HET_RATE">--LOSS_OF_HET_RATE </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>The rate at which a heterozygous genotype in a normal sample turns into a homozygous (via loss of heterozygosity) in the tumor (model assumes independent events, so this needs to be larger than reality).</b><br />
			
		</p>
		<p>
		<span class="label label-info ">double</span>
			&nbsp;<span class="label">0.5</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MATRIX_OUTPUT">--MATRIX_OUTPUT </a>
			 / <small>-MO</small>
		</h3>
		<p class="args">
			<b>Optional output file to write matrix of LOD scores to. This is less informative than the metrics output and only contains Normal-Normal LOD score (i.e. doesn't account for Loss of Heterozygosity). It is however sometimes easier to use visually.</b><br />
			
		</p>
			<p><b>Exclusion:</b> This argument cannot be used at the same time as <code>SECOND_INPUT</code>.</p>
		<p>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MAX_EFFECT_OF_EACH_HAPLOTYPE_BLOCK">--MAX_EFFECT_OF_EACH_HAPLOTYPE_BLOCK </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Maximal effect of any single haplotype block on outcome (-log10 of maximal likelihood difference between the different values for the three possible genotypes).</b><br />
			
		</p>
		<p>
		<span class="label label-info ">double</span>
			&nbsp;<span class="label">3.0</span>
			&nbsp;<span class="label label-warning">[ [ 0</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MAX_RECORDS_IN_RAM">--MAX_RECORDS_IN_RAM </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>When writing files that need to be sorted, this will specify the number of records stored in RAM before spilling to disk. Increasing this number reduces the number of file handles needed to sort the file, and increases the amount of RAM needed.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Integer</span>
			&nbsp;<span class="label">500000</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--NUM_THREADS">--NUM_THREADS </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>The number of threads to use to process files and generate fingerprints.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">1</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--OUTPUT">--OUTPUT </a>
			 / <small>-O</small>
		</h3>
		<p class="args">
			<b>Optional output file to write metrics to. Default is to write to stdout.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--OUTPUT_ERRORS_ONLY">--OUTPUT_ERRORS_ONLY </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>If true, then only groups that do not relate to each other as expected will have their LODs reported.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--QUIET">--QUIET </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Whether to suppress job-summary info on System.err.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--REFERENCE_SEQUENCE">--REFERENCE_SEQUENCE </a>
			 / <small>-R</small>
		</h3>
		<p class="args">
			<b>Reference sequence file.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--REQUIRE_INDEX_FILES">--REQUIRE_INDEX_FILES </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>A boolean value to determine whether input files should only be parsed if index files are available. Without turning this option on, the tool will need to read through the entirety of input files without index files either provided via the INPUT_INDEX_MAP or locally accessible relative to the input, which significantly increases runtime. If set to true and no index is found for a file, an exception will be thrown. This applies for both the INPUT and SECOND_INPUT files.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--SAMPLE_INDIVIDUAL_MAP">--SAMPLE_INDIVIDUAL_MAP </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>A tsv with two columns representing the individual with which each sample is associated.  The first column is the sample id, and the second column is the associated individual id.  Values in the first column must be unique. If INPUT_SAMPLE_MAP or SECOND_INPUT_SAMPLE_MAP is also specified, then the values in the first column of this file should be the sample aliases specified in the second columns of INPUT_SAMPLE_MAP and SECOND_INPUT_SAMPLE_MAP, respectively.  When this input is specified, expectations for matches will be based on the equality or inequality of the individual ids associated with two samples, as opposed to the sample ids themselves.  Samples which are not listed in this file will have their sample id used as their individual id, for the purposes of match expectations.  This means that one sample id could be used as the individual id for another sample, but not included in the map itself, and these two samples would be considered to have come from the same individual.  Note that use of this parameter only affects labelling of matches and mismatches as EXPECTED or UNEXPECTED.  It has no affect on how data is grouped for crosschecking.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--SECOND_INPUT">--SECOND_INPUT </a>
			 / <small>-SI</small>
		</h3>
		<p class="args">
			<b>A second set of input files (or lists of files) with which to compare fingerprints. If this option is provided the tool compares each sample in INPUT with the sample from SECOND_INPUT that has the same sample ID. In addition, data will be grouped by SAMPLE regardless of the value of CROSSCHECK_BY. When operating in this mode, each sample in INPUT must also have a corresponding sample in SECOND_INPUT. If this is violated, the tool will proceed to check the matching samples, but report the missing samples and return a non-zero error-code.</b><br />
			
		</p>
			<p><b>Exclusion:</b> This argument cannot be used at the same time as <code>MATRIX_OUTPUT</code>.</p>
		<p>
		<span class="label label-info ">List[String]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--SECOND_INPUT_INDEX_MAP">--SECOND_INPUT_INDEX_MAP </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>A tsv with two columns and no header which maps the second input files to corresponding indices; to be used when index files are not located next to second input files. First column must match the list of second inputs. </b><br />
			
		</p>
		<p>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--SECOND_INPUT_SAMPLE_MAP">--SECOND_INPUT_SAMPLE_MAP </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>A tsv with two columns representing the sample as it appears in the SECOND_INPUT data (in column 1) and the sample as it should be used for comparisons to INPUT (in the second column). Note that in case of unrolling files (file-of-filenames) one would need to reference the final file, i.e. the file that contains the genomic data. Need only include the samples that change. Values in column 1 should be unique. Values in column 2 should be unique even in union with the remaining unmapped samples. Should only be used with SECOND_INPUT. </b><br />
			
		</p>
		<p>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--showHidden">--showHidden </a>
			 / <small>-showHidden</small>
		</h3>
		<p class="args">
			<b>display hidden arguments</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--TMP_DIR">--TMP_DIR </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>One or more directories with space available to be used by this program for temporary storage of working files</b><br />
			
		</p>
		<p>
		<span class="label label-info ">List[File]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--USE_JDK_DEFLATER">--USE_JDK_DEFLATER </a>
			 / <small>-use_jdk_deflater</small>
		</h3>
		<p class="args">
			<b>Use the JDK Deflater instead of the Intel Deflater for writing compressed output</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--USE_JDK_INFLATER">--USE_JDK_INFLATER </a>
			 / <small>-use_jdk_inflater</small>
		</h3>
		<p class="args">
			<b>Use the JDK Inflater instead of the Intel Inflater for reading compressed input</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--VALIDATION_STRINGENCY">--VALIDATION_STRINGENCY </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Validation stringency for all SAM files read by this program.  Setting stringency to SILENT can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded.</b><br />
			
		</p>
			<p>
				The --VALIDATION_STRINGENCY argument is an enumerated type (ValidationStringency), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">STRICT</dt>
					<dd class="enum"></dd>
					<dt class="enum">LENIENT</dt>
					<dd class="enum"></dd>
					<dt class="enum">SILENT</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">ValidationStringency</span>
			&nbsp;<span class="label">STRICT</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--VERBOSITY">--VERBOSITY </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Control verbosity of logging.</b><br />
			
		</p>
			<p>
				The --VERBOSITY argument is an enumerated type (LogLevel), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">ERROR</dt>
					<dd class="enum"></dd>
					<dt class="enum">WARNING</dt>
					<dd class="enum"></dd>
					<dt class="enum">INFO</dt>
					<dd class="enum"></dd>
					<dt class="enum">DEBUG</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">LogLevel</span>
			&nbsp;<span class="label">INFO</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--version">--version </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>display the version number for this tool</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>

        <hr>
        <p><a href='#top'><i class='fa fa-chevron-up'></i> Return to top</a></p>
        <hr>
        <p class="see-also">See also 
        	<a href="https://www.broadinstitute.org/picard/guide/">Picard Documentation Index</a> |
        	<a href="index.html">Tool Docs Index</a> |
        	<a href="http://gatkforums.broadinstitute.org/">Support Forum</a>
        </p>

        <p class="version">Picard version 3.0.0-17-g83b016e7f-SNAPSHOT built at 27-41-2023 01:41:02.
        </p>

	</div>

	<?php printFooter($module); ?>